---
title: "Cell type annotation"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(Seurat)
library(tidyverse)
library(viridis)
library(SingleR)
library(AnnotationHub)
library(ggsci)
library(ggalluvial)
source("./code/plotting.R")
source("./code/seurat_sidekick.R")
seu_s <- readRDS("./output/DG_linnarson/seu_s.rds")
seu_u <- readRDS("./output/DG_linnarson/seu_u.rds")
markers <- readRDS("./output/DG_linnarson/markers_LR.rds")
```

```{r}
ah <- AnnotationHub()
edb <- ah[["AH73905"]]
gns <- select(edb, keys = keys(edb, keytype = "GENEID"), keytype = "GENEID",
              columns = c("GENEIDVERSION","DESCRIPTION", "SYMBOL", "ENTREZID"))
gns <- gns[,c(2,4,3,5)]
names(gns) <- c("gene", "gene_name", "description", "entrezgene")
```

# Using reference
```{r}
data("mouse.rnaseq")
```

```{r}
# Use Ensembl gene ID in reference
ref_use <- mouse.rnaseq$data
rownames(ref_use) <- gns$gene[match(rownames(ref_use), gns$gene_name)]
ref_use <- ref_use[!is.na(rownames(ref_use)),]
```

```{r}
annot <- SingleR("single", GetAssayData(seu_s), ref_data = ref_use,
                 types = mouse.rnaseq$types)
annot_u <- SingleR("single", GetAssayData(seu_u), ref_data = ref_use,
                 types = mouse.rnaseq$types)
```

```{r}
mean(annot$pval < 0.05)
mean(annot_u$pval < 0.05)
```

```{r}
summary(annot_u$pval[annot_u$pval > 0.05])
```

So I should take the annotation from the unspliced matrix with a grain of salt, though the p-value doesn't seem that bad. Well, to be honest, largely due to all those extreme p-values in differential expression even of genes that don't look like more expressed in the cluster of interest, I no longer trust p-values. 
```{r}
seu_s <- AddMetaData(seu_s, setNames(annot$labels, annot$cell.names), col.name = "cell_type")
seu_u <- AddMetaData(seu_u, setNames(annot_u$labels, annot_u$cell.names), 
                     col.name = "cell_type")
```

```{r}
saveRDS(seu_s, "./output/DG_linnarson/seu_s.rds")
saveRDS(seu_u, "./output/DG_linnarson/seu_u.rds")
```

I want matching colors for cell type annotations.
```{r}
colors_use <- setNames(pal_d3("category20")(17), 
                       sort(union(unique(annot$labels), unique(annot_u$labels))))
```

```{r}
DimPlot(seu_s, reduction = "tsne", group.by = "cell_type", pt.size = 0.5) +
  scale_color_manual(values = colors_use)
```

After removing the doublets, ridiculous things like cardiomyocytes and adipocytes are gone, so those were probably doublets
```{r}
DimPlot(seu_u, reduction = "tsne", group.by = "cell_type", pt.size = 0.5) +
  scale_color_manual(values = colors_use)
```

It seems that more cells get labeled neurons and fewer cells get labeled NPCs in the unspliced matrix than the spliced matrix. Plot an alluvial plot to see where the cell types are going.
```{r}
df <- tibble(barcode = Cells(seu_s),
             spliced = seu_s$cell_type,
             unspliced = seu_u$cell_type) %>% 
  dplyr::count(spliced, unspliced)
```

```{r, fig.height=10, fig.width=8}
ggplot(df, aes(y = n, axis1 = spliced, axis2 = unspliced)) +
  geom_stratum() +
  geom_alluvium(aes(fill = spliced)) +
  geom_text(stat = "stratum", label.strata = TRUE) +
  scale_x_discrete(limits = c("spliced", "unspliced"), expand = c(0,0)) +
  scale_y_continuous(expand = c(0,0)) +
  scale_fill_manual(values = colors_use)
```

So yes, many cells labeled NPCs in the spliced matrix become neurons in the unsplicedd matrix, as expected. Most cells labeled qNSCs and some NPCs in the spliced matrix become astrocytes in the unspliced matrix. Most aNSCs became neurons, and monocytes became microglia. Epithelial cells remain epithelial cells. It's hard to read the smaller cell populations in this plot, so I'll make another plot without the neurons and NPCs. 

```{r, fig.width=7, fig.height=8}
df %>% 
  dplyr::filter(!spliced %in% c("Neurons", "NPCs", "qNSCs")) %>% 
  ggplot(aes(y = n, axis1 = spliced, axis2 = unspliced)) +
  geom_stratum() +
  geom_alluvium(aes(fill = spliced)) +
  geom_text(stat = "stratum", label.strata = TRUE) +
  scale_x_discrete(limits = c("spliced", "unspliced"), expand = c(0,0)) +
  scale_y_continuous(expand = c(0,0)) +
  scale_fill_manual(values = colors_use)
```

So many aNSCs became neurons, but so did some endothelial cells and some oligodendrocytes and some fibroblasts. Anyway, take it with a grain of salt. Perhaps these are doublets that I failed to remove? Most monocytes in the spliced matrix become microglia in the unspliced matrix, and that's fine since microglia are macrophages. 

# Using marker genes
```{r}
DimPlot(seu_s, reduction = "tsne", group.by = "concat_clusters", pt.size = 0.5, label = TRUE)
```

```{r}
DimPlot(seu_u, reduction = "tsne", group.by = "concat_clusters", pt.size = 0.5, label = TRUE)
```

I think the largest clusters are mature granule cells. Cluster 20 seems like a population soon to become different from most mature GCs.

## Cluster 26

```{r}
markers %>% 
  dplyr::filter(cluster == 26, p_val_adj < 0.05)
```

What I know is that Sirt2 is relevant to Alzheimer's (https://www.ncbi.nlm.nih.gov/pubmed/30871086/) and cancer. Bcas1 is a candidate oncogene. According to bulk RNA-seq data from NCBI's gene page, it's not only highly expressed in the brain, but also in the stomach, small intestine, and colon. Bcas1 is also found to be a myelenation protein found specifically in oligodendrocytes and schwann cells (https://www.ncbi.nlm.nih.gov/pubmed/28230289/). So I suspect that cluster 26 might be  oligodendrocytes, though cluster 23 is also oligodendrocytes. Ust plays a role in neuronal migration (https://www.ncbi.nlm.nih.gov/pubmed/18819920/). Bmp4? Doesn't sound that relevant to the brain, right? It was said that mice lacking Enpp6 have myelin sheath abnormalities. SingleR labeled cluster 26 oligodendrocytes, and the neighboring cluster 15 aNSCs. I wonder what makes this cluster different from cluster 23.

```{r}
FeaturePlot(seu_s, "ENSMUSG00000015149.14", pt.size = 0.5)
```

## Cluster 25
```{r}
markers %>% 
  dplyr::filter(cluster == 25, p_val_adj < 0.05)
```

Obviously, cluster 25 is erythrocytes. Somehow SingleR was stupid enough to label it neurons.

```{r}
FeaturePlot(seu_s, "ENSMUSG00000069919.7", pt.size = 0.5)
```

## Cluster 24
```{r}
markers %>% 
  dplyr::filter(cluster == 24, p_val_adj < 0.05)
```

Hmgb2 mediates transition from quiescent to activated adult neural stem cells (https://www.ncbi.nlm.nih.gov/pubmed/28771884/). So cluster 24 must be neural stem cells primed for activation, and that explains its location on tSNE. Spc24 is involved in cancer and is associated with lipid level in GWAS. I also saw centromere protein M and histone. Cenpm encodes a protein that binds to the spindle microtubules, so this supports labeling cluster 24 as active adult neural stem cells. Well, or maybe neuronal intermediate progenitor cells?

```{r}
FeaturePlot(seu_s, "ENSMUSG00000054717.7", pt.size = 0.5)
```

## Cluster 23
```{r}
markers %>% 
  dplyr::filter(cluster == 23, p_val_adj < 0.05)
```

Yep, with all those myelin genes, cluster 23 is obviously oligodendralcytes. 

```{r}
FeaturePlot(seu_s, "ENSMUSG00000031425.15", pt.size = 0.5)
```

Cluster 26 also expresses this gene, so perhaps I was right in saying that it's oligodendrocytes.

## Cluster 22
```{r}
markers %>% 
  dplyr::filter(cluster == 22, p_val_adj < 0.05)
```

Reelin - Cajalâ€“Retzius cells. 

```{r}
FeaturePlot(seu_s, "ENSMUSG00000042453.14", pt.size = 0.5)
```

## Cluster 21
```{r}
markers %>% 
  dplyr::filter(cluster == 21, p_val_adj < 0.05)
```

SingleR thinks cluster 21 is neurons. But what kind of neurons? Again, Khdrbs3 is related to cancer and conntrols mouse hippocampal synapse specification (https://www.ncbi.nlm.nih.gov/pubmed/27174676). Mgat4c is related to HIV, but otherwise not much is known about it; it's much more overexpressed in the unspliced matrix. Ywhag regulates neuronal differentiation in the hippocampus (https://www.ncbi.nlm.nih.gov/pubmed/28412242/). It's also invoved in migration of pyramidal neurons in the developing cerebral cortex (https://www.ncbi.nlm.nih.gov/pubmed/27288018/, https://www.ncbi.nlm.nih.gov/pubmed/26297819/). This gene also seems to be involved in cancer. In mice, knocking down this gene impairs memory (https://www.jneurosci.org/content/34/14/4801). In rat, this gene is expressed in hippocampal pyramidal neurons and dentate granule cells (https://doi.org/10.1016/j.brainres.2011.06.036). I also saw many genes related to synapses. So it makes sense to say this cluster is neurons of some sort. But why don't those granule cells express these genes? I suspect that this cluster is CA3 pyramidal neurons, since some, though not all, of CA3 pyramidal neuron markers (from https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5554463/) highlight this cluster.

```{r}
FeaturePlot(seu_s, "ENSMUSG00000022332.8", pt.size = 0.5)
```

Here's one of those known CA3 pyramidal neuron markers:
```{r}
FeaturePlot(seu_s, gns$gene[gns$gene_name == "Bok"], pt.size = 0.5)
```

## Cluster 20
```{r}
markers %>% 
  dplyr::filter(cluster == 20, p_val_adj < 0.05)
```

I'm pretty sure this is some sort of neuron, though I can't tell between this cluster and other neuronal clusters with these marker genes.
```{r}
FeaturePlot(seu_s, "ENSMUSG00000057897.14", pt.size = 0.5)
```

So what differentiates this cluster from most other neurons?

```{r}
main_neu_clusts <- c(1,2,4,5,8,9,17)
markers20 <- FindMarkers_clusts(seu_s, seu_u, clusts.1 = 20, clusts.2 = main_neu_clusts,
                                latent.vars = c("nCount_SCT", "nFeature_SCT"), gns = gns)
markers20
```

```{r}
FeaturePlot(seu_s, "ENSMUSG00000037679.9", pt.size = 0.5)
```

## Cluster 19
```{r}
markers %>% 
  dplyr::filter(cluster == 19, p_val_adj < 0.05)
```

Several top DE genes have something to do with glutamate. There are also DE genes for voltage gated channels and synapses. Gad2 and Gad1 are GABAergic markers, used to synthesize GABA. So I suppose this cluster is GABAergic neurons.
```{r}
FeaturePlot(seu_s, "ENSMUSG00000070880.10", pt.size = 0.5)
```

## Cluster 18
```{r}
markers %>% 
  dplyr::filter(cluster == 18, p_val_adj < 0.05)
```

The problem with this cluster is that the DE genes are not highly expressed even within the cluster, making it more difficult to infer cell type from DE genes. 
Lgals1 is said to be involved in cell-cell and cell-matrix interactions. Serpinh1 is involved in collagen synthesis. Dab2 is said to be a tumor suppressor. 

```{r}
FeaturePlot(seu_s, "ENSMUSG00000068220.6", pt.size = 0.5)
```

Cluster 18 is split into multiple parts on this plot, and this gene highlight the left part, which is labeled as fibroblasts by SingleR. 

## Cluster 17
```{r}
markers %>% 
  dplyr::filter(cluster == 17, p_val_adj < 0.05)
```

This is one of those clusters among the granule cells that Leiden thinks is a subtype, though it's hard to find markers that cleanly distinguish this cluster from other granule cells. I don't know what to make of this kind of clusters, but hopefully this will become clear when I look at the spatial data.

```{r}
main_neu_clusts <- c(1,2,4,5,8,9,20)
markers17 <- FindMarkers_clusts(seu_s, seu_u, clusts.1 = 17, clusts.2 = main_neu_clusts,
                                latent.vars = c("nCount_SCT", "nFeature_SCT"), gns = gns)
markers17
```

```{r}
FeaturePlot(seu_s, "ENSMUSG00000032503.18", pt.size = 0.5)
```

## Cluster 16
```{r}
markers %>% 
  dplyr::filter(cluster == 16, p_val_adj < 0.05)
```

I expect this cluster to be something like more immature granule cells. Or maybe not, since the root part of that granule cell thing has two branches, and cluster 16 is part of the other branch distinct from the main granule cell thing. It seems that Zbtb20 is involved in either neurogenesis or glialgenesis, depending on the context. It also promotes astrocytogenesis in the mouse neocortex and is expressed in late stage NPCs (https://www.ncbi.nlm.nih.gov/pubmed/27000654). However, it seems that this gene does not distinguish between this cluster annd the other branch of NPCs. 

```{r}
FeaturePlot(seu_s, "ENSMUSG00000033209.17", pt.size = 0.5)
```

## Cluster 15
```{r}
markers %>% 
  dplyr::filter(cluster == 15, p_val_adj < 0.05)
```

SingleR thinks this cluster is aNSCs. But really? Olig1 is required for oligodendrocyte differentiation. It's expressed in both this cluster and the other oligodendrocyte cluster. This makes me suspect that this cluster is actually OPCs. Pdgfra is downregulated, as there're much fewer unspliced counts than spliced counts. This has been validated experimentally, that it's downregulated as OPCs differentiate: https://doi.org/10.1016/j.devcel.2018.06.025. That's more evidence that this cluster is OPCs, as this gene is not detected in the other oligodendrocyte cluster. To be honest, now I kind of wonder if manual cell type annotation is really better, since it feels like a judgmental call based on a minority of DE genes for which some relevant literature can be found. Many DE genes are not well studied, and some are expressed in some other clusters of a supposedly very different cell type besides this one.

```{r}
FeaturePlot(seu_s, "ENSMUSG00000035131.14", pt.size = 0.5)
```

## Cluster 14
SingleR says this cluster is endothelial cells.
```{r}
markers %>% 
  dplyr::filter(cluster == 14, p_val_adj < 0.05)
```

Flt1 coordinates angiogenesis (https://www.ncbi.nlm.nih.gov/pubmed/27142980/). Egfl7 is a marker of endothelial lineage (https://www.ncbi.nlm.nih.gov/pubmed/24740971/). So yeah, I agree that this cluster is endothelial cells.

```{r}
FeaturePlot(seu_s, "ENSMUSG00000029648.13", pt.size = 0.5)
```

## Cluster 13
SingleR thinks cluster 13 is NPCs and neurons
```{r}
markers %>% 
  dplyr::filter(cluster == 13, p_val_adj < 0.05)
```

```{r}
other_NPCs <- c(11, 6)
markers13 <- FindMarkers_clusts(seu_s, seu_u, clusts.1 = 13, clusts.2 = other_NPCs,
                                latent.vars = c("nCount_SCT", "nFeature_SCT"), gns = gns)
markers13
```

```{r}
FeaturePlot(seu_s, gns$gene[gns$gene_name == "Tubb2a"], pt.size = 0.5)
```

Top DE genes include tubulin, which is expected for neurons making axons. But again, those DE genes from this cluster vs. all other cells can't distinguish between this cluster and the other branch of NPCs. DE genes between this cluster and the other branch of NPCs exist, but I don't know what this says about this cluster. Well, it seems that some of the top enriched DE genes between cluster 13 and the other NPC branch are also more highly expressed in mature granule cells. Top depleted DE genes are more expressed in clusters 11 and 6, but not the other mature granule cells. Perhaps all that this says is that cluster 13 is a later stage in differentiation, and cluster 11 is an earlier stage. Then it made two branches, one for the granule cells, and another for this cluster. I look forward to transfering the cluster labels to spatial data and see.

## Cluster 12
This cluster is detached from the main granule cell cluster. SingleR thinks it's mainly neurons.
```{r}
markers %>% 
  dplyr::filter(cluster == 12, p_val_adj < 0.05)
```

Some of the top DE genes are mitochondrial genes for the electron transport chain. This is not the case for the other clusters I inspected, so I don't think it's spurious. However, upon visual inspection, those mitochondrial genes don't seem much more highly expressed than in other cells. Some of the top DE genes are also expressed by the mature granule cells and the CA3 pyramidal neurons, but not NPCs, astrocytes, oligodendrocytes, and the other non-neuronal clusters. So I suppose this cluster is probably some sort of mature neuron.
```{r}
FeaturePlot(seu_s, "ENSMUSG00000026833.18", pt.size = 0.5)
```

How does this cluster differ from the other neurons?

```{r}
other_neus <- c(1,2,4,5,8,9,17,20,21)
markers12 <- FindMarkers_clusts(seu_s, seu_u, clusts.1 = 12, clusts.2 = other_neus,
                                latent.vars = c("nCount_SCT", "nFeature_SCT"), gns = gns)
markers12
```

These are mostly genes more expressed in the granule cells, except those electron transport chain genes. So somehow these neurons do a lot of aerobic respiration.
```{r}
FeaturePlot(seu_s, "ENSMUSG00000090223.2", pt.size = 0.5)
```

## Cluster 11
I'm pretty sure it's early in differentiation. 
```{r}
markers %>% 
  dplyr::filter(cluster == 11, p_val_adj < 0.05)
```

Igfbpl1 and Sox11 are among the neuroblast2 markers in the Hochgerner paper.
```{r}
FeaturePlot(seu_s, gns$gene[gns$gene_name == "Calb2"], pt.size = 0.5)
```

This is one of the neuroblast1 markers from the Hochgerner paper, and it pretty nicely highlights cluster 11. So I think this cluster must be neuroblasts. Igfbpl1 and Sox11 also highlight clusters 6 annd 16, so those must be later stage neuroblasts. It was said that these cells migrate to the granule cell layer and mature, so I'll watch for these cells in the spatial data.

## Cluster 10
```{r}
markers %>% 
  dplyr::filter(cluster == 10, p_val_adj < 0.05)
```

Hopx is a human basal radial glial cell marker. It's also expressed in dentage gyrus subgranular layer's radial astrocyte stem cells, which eventually would differentiate into granular cells (https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2416439/). Msi2 seems to be related to stem cell quiescence. So I think cluster 10 is radial astrocyte stem cells.

```{r}
FeaturePlot(seu_s, gns$gene[gns$gene_name == "Hopx"], pt.size = 0.5)
```

Some of the top DE genes are highly expressed in both clusters 10 and 3 (which I think is astrocytes). 

## Cluster 9
```{r}
markers %>% 
  dplyr::filter(cluster == 9, p_val_adj < 0.05)
```

```{r}
FeaturePlot(seu_u, gns$gene[gns$gene_name == "Cebpd"], pt.size = 0.5)
```

Again, one of those granule cell clusters.

## Cluster 8
```{r}
markers %>% 
  dplyr::filter(cluster == 8, p_val_adj < 0.05)
```

I rediscovered a immature granule cell marker in the Hochgerner paper -- Fxyd7. So I suppose this cluster is immature granule cells.
```{r}
FeaturePlot(seu_s, gns$gene[gns$gene_name == "Fxyd7"], pt.size = 0.5)
```

This gene also highlights cluster 2.

## Cluster 7
SingleR thinks cluster 7 is microglia.

```{r}
markers %>% 
  dplyr::filter(cluster == 7, p_val_adj < 0.05)
```

Lgmn is said to be expressed in macrophages and is involved in lysosomes, so I think SingleR is right.
```{r}
FeaturePlot(seu_s, gns$gene[gns$gene_name == "Lgmn"], pt.size = 0.5)
```

## Cluster 6
I think this is neuroblast or an earlier stage of immature GCs.
```{r}
markers %>% 
  dplyr::filter(cluster == 6, p_val_adj < 0.05)
```

I got Zbtb20 again, which is also highly expressed in cluster 16. I also got Sox11 and Igfbpl1 again, so this cluster is also what Hochgerner would call neuroblast2. So here I do seem to have two different groups of later stage neuroblasts. I get some Rpl/s genes showing up as markers for neuroblasts and immature GCs. I still remember how so many genes that differ between kallisto and CellRanger are Rpl/s. So I kind of wonder if those genes really are technical artifacts. 
```{r}
FeaturePlot(seu_s, gns$gene[gns$gene_name == "Gm17750"], pt.size = 0.5)
```

## Cluster 5, 4, 1
One of those mature GC cluster again. Just too lazy to find out how it differs from other mature GCs clusters.

## Cluster 3
Pretty sure it's astrocytes.
```{r}
markers %>% 
  dplyr::filter(cluster == 3, p_val_adj < 0.05)
```

Slc1a2 is a glutamate transport in astrocytes.
```{r}
FeaturePlot(seu_s, gns$gene[gns$gene_name == "Slc1a2"], pt.size = 0.5)
```

Cassical astrocyte markers:

```{r}
astrocyte_markers <- c("Gfap", "Hes5", "Sox9")
FeaturePlot(seu_s, gns$gene[gns$gene_name %in% astrocyte_markers], pt.size = 0.5)
```

I still wander if something is wrong with this dataset or how I processed it since some of the classical markers are absent or not so highly expressed.

## Cluster 2
I think it's a later stage immature GC cluster.
```{r}
markers %>% 
  dplyr::filter(cluster == 2, p_val_adj < 0.05)
```

```{r}
FeaturePlot(seu_s, gns$gene[gns$gene_name == "Lrp1b"], pt.size = 0.5)
```

## Plot manual annotation
```{r}
manual_annot <- tribble(~ concat_clusters, ~ cell_type_manual,
                        26, "Oligodendrocytes2",
                        25, "Erythrocytes",
                        24, "nIPCs",
                        23, "Oligodendrocytes1",
                        22, "Cajalâ€“Retzius",
                        21, "CA3 pyramidal",
                        20, "Mature GCs 1", # The number does not signal sequence in differentiation
                        19, "GABAergic",
                        18, "Fibroblasts?",
                        17, "Mature GCs 2",
                        16, "Neuroblasts3",
                        15, "OPCs",
                        14, "Endothelial",
                        13, "Neuroblasts4",
                        12, "Neurons1",
                        11, "Neuroblasts1",
                        10, "RGL",
                        9, "Mature GCs 3",
                        8, "Immature GCs 1",
                        7, "Microglia",
                        6, "Neuroblasts2",
                        5, "Mature GCs 4",
                        4, "Mature GCs 5",
                        3, "Astrocytes",
                        2, "Immature GCs 2",
                        1, "Mature GCs 6")
manual_annot$concat_clusters <- factor(manual_annot$concat_clusters, levels = 1:26)
seu_s@meta.data <- left_join(seu_s@meta.data, manual_annot, by = "concat_clusters")
rownames(seu_s@meta.data) <- Cells(seu_s)
seu_u$cell_type_manual <- seu_s$cell_type_manual
```

```{r}
DimPlot(seu_s, reduction = "tsne", group.by = "cell_type_manual", pt.size = 0.5, label = TRUE) +
  theme(legend.position = "none")
```

```{r}
DimPlot(seu_u, reduction = "tsne", group.by = "cell_type_manual", pt.size = 0.5, label = TRUE) +
  theme(legend.position = "none")
```

```{r}
saveRDS(seu_s, file = "./output/DG_linnarson/seu_s.rds")
saveRDS(seu_u, "./output/DG_linnarson/seu_u.rds")
saveRDS(gns, "./output/DG_linnarson/gene_names_97.rds")
```
