---
title: "Microdissection"
author: "Lambda Moses"
date: "`r Sys.Date()`"
output: rmarkdown::html_vignette
vignette: >
  %\VignetteIndexEntry{Microdissection}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Here I explore the history and geography of spatial transcriptomics based on microdissection through the metadata I collected.
```{r}
library(museumst)
library(ggplot2)
library(purrr)
library(dplyr)
library(sf)
theme_set(theme_bw())
```

# Import data
By default, the `read_metadata` function reads the appropriate sheet from a cache within this package. This may be outdated, since the Google Sheets are constantly updated. Use the argument `update = TRUE` to download the most up to date version. The default is `FALSE` to avoid API limits during automated CRAN checks.
```{r}
sheet <- read_metadata("Microdissection")
```

# Number of publications
How many publications per year?
```{r}
# Remove duplicates as different datasets in the same publication have their own rows
publications <- get_pubs_df(sheet, other_cols = c("repo", "accession"))
```

How many publications are there in total in this sheet?
```{r}
nrow(publications)
```

```{r}
# For maps later in this notebook
city_gc <- geocode_inst_city(publications)
pubs_on_map2 <- partial(pubs_on_map, city_gc = city_gc)
```

## Overall
Number of publications per year. Preprints are excluded from plots over years as it takes months to publish so the dates of the preprints are incoherent with those for the other papers. 
```{r}
events <- read_major_events()
```

```{r, fig.width=8, fig.height=7}
events %>% 
  filter(category == "microdissection") %>% 
  plot_timeline(c(0.3, -0.3, 0.5, -0.3, 0.3, -0.7, 1, -0.9, 0.5), 
                expand_x = c(0.1, 0.1), expand_y = c(0.05, 0.05))
```

By default, preprints are excluded from plots of number of publications over time, since the timing of preprints is incoherent to that of published papers. Including preprints will inflate the number of publications in recent months, and if we make the same plot a few months later, that inflation we see now will be gone and moved to a more recent date. You can add the `preprints = TRUE` argument to all functions plotting things over time in this package to include preprints. You can also use the `binwidth` argument to specify bin width in days. The default is 365 days.

```{r}
pubs_per_year(publications)
```

## By method
How about when broken down by method (not for data analysis sheets)?

Get this into another R Markdown file since the analysis ones don't have the method column. For LCM, this should be taken with a grain of salt, since I don't think my collection of LCM papers is complete. LCM started far earlier than this. Because even I don't believe this myself, I don't think this figure will go into the paper. 

I'm not sure if many of those papers even count as spatial since they simply used H&E to get a ROI, use LCM to isolate it for RNA-seq, that's it, sort of forgetting about the spatial part. But Niche-seq is also used this way and those who made Niche-seq called it spatial. I guess it's just that you know where the sample comes from in the tissue and what's next to it. Well, suppose you say, "I'm visiting LA. Not just LA, but more specifically Little Bangladesh." Is that spatial? I know where Little Bangladesh is and what's around it. I don't think geostatisticians will consider that statement spatial, because it does not use spatial statistics, just like you can fit non-spatial linear regression models to data from different neighborhoods though you know where those neighborhoods are. I suppose it's just an example of how the same word has different meanings in different context, and I'll stick to the LCM context since so far, spatial statistics has not be all that widely used in transcriptomics.

```{r}
methods <- unnest_cat(sheet, method)
```

```{r, fig.width=6, fig.height=12}
pubs_per_year(methods, facet_by = "method")
```

I don't know how much this actually says due to the small sample size. 

## By species

```{r}
species <- unnest_cat(sheet, species)
```

```{r}
pubs_per_cat(species, species)
```

```{r}
pubs_per_cat(species, species, n_top = 5, isotype = TRUE, img_unit = 2)
```

## By journal
```{r}
sort(table(publications$journal))
```

## Location
### General
Just some barplots for number of publications per institution, city, and country. 
```{r}
pubs_per_cat(publications, country)
```

How about per capita? Actually this probably does not say much since many researchers in the West are immigrants. Then it raises the burning political question of who counts as the "population".

```{r}
pubs_per_capita(publications, plot = "bar")
```

How about country over time? There might not be enough publications to show a trend.
```{r, fig.height=12, fig.width=6}
pubs_per_year(publications, facet_by = "country")
```
Now look at cities. 

```{r}
pubs_per_cat(publications, city)
```

Institutions
```{r}
pubs_per_cat(publications, institution)
```

12345678, U, C, L, A, UCLA fight fight fight! We beat 'SC in this plot!

OK, now here comes the maps!

### Worldwide
```{r}
pubs_on_map2(publications)
```

Let me also plot the per capita thing on a map, as a choropleth
```{r}
pubs_per_capita(publications)
```

Break down by species
```{r, fig.width=8, fig.height=8}
pubs_on_map2(species, facet_by = "species")
```


Probably this does not tell us much since there's only 1 study for some species. Again, we see the received wisdom that research is mostly confined to the West, especially Western Europe, west coast of the US, and New England, but not as much in the other parts of the "West".

By method
```{r, fig.width=8, fig.height=8}
pubs_on_map2(methods, facet_by = "method")
```

I suppose this merely tells us where those techniques were invented and how they spread. For instance, voxelation of the brain did not spread outside UCLA to my best knowledge, while Tomo-seq and Niche-seq sort of spread beyond the lab where they originated. 

### Europe
Plot a map just for Europe. A problem here is that if I simply filter the `world` sf object, I'll get some islands away from what we usually think is Europe. Alas, the sad legacy of colonialism! Kind of feel bad for Russians since Russia is so large that it makes the map look bad and I don't have a paper from Russia in this spreadsheet. 

```{r}
pubs_on_map2(publications, zoom = "europe")
```

```{r}
pubs_per_capita(publications, "europe")
```

I didn't realize that Switzerland and Netherlands were pretty great. Again, this is a small sample size, so I take this with a grain of salt until I plot the other sheets. From the other sheets, I already know that Sweden, the UK, and Germany have many contributions. 

```{r, fig.width=6, fig.height=8}
pubs_on_map2(species, zoom = "europe", facet_by = "species")
```

Maybe I just made this plot for fun. Not sure what to say about it.

```{r, fig.width=8, fig.height=6}
pubs_on_map2(methods, zoom = "europe", facet_by = "method")
```

### USA
Also a plot just for America. I did not encounter a publication in this spreadsheet that is from Hawaii or Alaska, but I'll include Hawaii and Alaska in the map anyway in case I get one in the future.

```{r}
pubs_on_map2(publications, zoom = "usa")
```

```{r}
pubs_per_capita(publications, "usa")
```

```{r, fig.width=8, fig.height=6}
pubs_on_map2(species, zoom = "usa", facet_by = "species")
```

```{r, fig.width=8, fig.height=6}
pubs_on_map2(methods, zoom = "usa", facet_by = "method")
```

# Word cloud
## Titles
```{r, fig.height=7, fig.width=7}
plot_wordcloud(sheet)
```

So not all use of the term "spatial" in the current era is from ST.

```{r, fig.height=7, fig.width=7}
plot_wordcloud(sheet, species_use = "Mus musculus", scale = c(5, 0.1))
```

```{r, fig.height=7, fig.width=7}
plot_wordcloud(sheet, species_use = "Homo sapiens", scale = c(5, 0.1))
```

## Tissues
```{r, fig.height=5, fig.width=5}
plot_wordcloud(sheet, col_use = "tissue")
```

```{r, fig.height=5, fig.width=5}
plot_wordcloud(sheet, species_use = "Mus musculus", col_use = "tissue")
```

```{r, fig.height=5, fig.width=5}
plot_wordcloud(sheet, species_use = "Homo sapiens", col_use = "tissue")
```

## Over time

```{r}
range(sheet$date_published)
```

```{r, fig.height=6, fig.width=6}
plot_wordcloud(sheet, year_min = 2002, year_max = 2010)
```

```{r, fig.height=7, fig.width=7}
plot_wordcloud(sheet, year_min = 2010, year_max = 2015)
```

```{r, fig.height=7, fig.width=7}
plot_wordcloud(sheet, year_min = 2015, year_max = 2021)
```

## Department names

```{r, fig.height=7, fig.width=7}
plot_wordcloud(sheet, col_use = "department", other_stop_words = inst_words)
```

## Downstream analyses
```{r, fig.height=5, fig.width=5}
plot_wordcloud(sheet, "downstream")
```

# Programming languages
```{r}
langs <- unnest_cat(sheet, language)
```

```{r}
pubs_per_cat(langs, language, isotype = TRUE, img_unit = 5)
```

# Data and code availability

How many publications have provided a code repo?
```{r}
hist_bool(publications, !is.na(repo))
```

How many publications have provided an accession for sequencing data?
```{r}
hist_bool(publications, !is.na(accession))
```

It seems that more recent publications are more likely to provide an accession. Is this significant? Here I fit a logistic regression model using date published to predict whether accession is available and test if the coefficients are 0.
```{r}
test_year_bool(publications, !is.na(repo))
```

```{r}
test_year_bool(publications, !is.na(accession))
```

So there sort of is a positive association between when the paper was published and whether accession is available, and more recent publications are more likely to have accessions.
