---
title: "smFISH"
author: "Lambda Moses"
date: "5/25/2020"
output: html_document
params:
  sheet: "smFISH"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

Here I explore the history and geography of spatial transcriptomics based on microdissection through the metadata I collected.
```{r}
library(tidyverse)
library(googlesheets4)
library(lubridate)
library(magick)
library(scales)
library(zeallot)
source("code/metadata_plotting.R")
theme_set(theme_bw())
```

# Import data
```{r}
gs4_deauth()
sheet <- read_sheet("https://docs.google.com/spreadsheets/d/1sJDb9B7AtYmfKv4-m8XR7uc3XXw_k4kGSout8cqZ8bY/edit#gid=566523154", sheet = params$sheet)
```

Add the year column as this will be used for plotting rather than the full date.
```{r}
sheet <- sheet %>% 
  mutate(year = year(date_published))
```

# Number of publications
How many publications per year?
```{r}
# Remove duplicates as different datasets in the same publication have their own rows
publications <- sheet %>% 
  select(year, title, journal, repo:department) %>% 
  distinct()
```

```{r}
anyDuplicated(publications$title)
```

How many publications are there in total in this sheet?
```{r}
nrow(publications)
```

```{r}
# For maps later in this notebook
c(inst_gc, city_gc) %<-% geocode_inst_city(publications)
pubs_on_map2 <- partial(pubs_on_map, inst_gc = inst_gc, city_gc = city_gc)
```

## Overall
Number of publications per year. Preprints are excluded from plots over years as it takes months to publish so the dates of the preprints are incoherent with those for the other papers. 
```{r}
gs4_deauth()
events <- read_sheet("https://docs.google.com/spreadsheets/d/1sJDb9B7AtYmfKv4-m8XR7uc3XXw_k4kGSout8cqZ8bY/edit#gid=566523154", sheet = "major events")
events$date_published <- as_date(events$date_published)
```

```{r, fig.width=5.5, fig.height=3}
events %>% 
  filter(category == "smFISH") %>% 
  plot_timeline(c(0.8, -0.5, 0.6, -0.5, 1, 0.6, -0.4, 0.7), 
                expand_x = c(0.1, 0.1), expand_y = c(0.05, 0.05))
```

Is it bad to borrow the original figures from those historical papers? Well, here I sort of want to frame then as museum exhibits so it's all the cooler to have the original rather than my parody.

```{r}
pubs_per_year(publications, break_width = 1)
```

## By method
How about when broken down by method (not for data analysis sheets)?

Get this into another R Markdown file since the analysis ones don't have the method column. For LCM, this should be taken with a grain of salt, since I don't think my collection of LCM papers is complete. LCM started far earlier than this. Because even I don't believe this myself, I don't think this figure will go into the paper. 

I'm not sure if many of those papers even count as spatial since they simply used H&E to get a ROI, use LCM to isolate it for RNA-seq, that's it, sort of forgetting about the spatial part. But Niche-seq is also used this way and those who made Niche-seq called it spatial. I guess it's just that you know where the sample comes from in the tissue and what's next to it. Well, suppose you say, "I'm visiting LA. Not just LA, but more specifically Little Bangladesh." Is that spatial? I know where Little Bangladesh is and what's around it. I don't think geostatisticians will consider that statement spatial, because it does not use spatial statistics, just like you can fit non-spatial linear regression models to data from different neighborhoods though you know where those neighborhoods are. I suppose it's just an example of how the same word has different meanings in different context, and I'll stick to the LCM context since so far, spatial statistics has not be all that widely used in transcriptomics.

```{r, include=params$datasets, eval=params$datasets}
methods <- sheet %>% 
  select(year, title, journal, method, country, institution) %>% 
  distinct()
```

```{r, fig.width=4, fig.height=8}
pubs_per_year(methods, facet_by = "method", break_width = 1)
```
I don't know how much this actually says due to the small sample size. This is also kind of confusing since Geo-seq really is a form of LCM coupled with SMART-seq, so it should be categorized as LCM, but the authors gave it a new name. 

## By species
```{r}
species_img <- readRDS("output/species_img.rds")
```

```{r}
species <- sheet %>% 
  select(year, title, species, country, institution) %>% 
  distinct()
```

```{r}
unique(species$species)
```

```{r}
pubs_per_cat(species, category = "species")
```

```{r}
top_species <- species %>% 
  count(species) %>%
  top_n(5, n) %>% 
  filter(!is.na(species)) %>% 
  arrange(desc(n)) %>% 
  pull(species)
species_img$image <- purrr::map(species_img$image_paths, image_read)
```

```{r}
species %>% 
  filter(species %in% top_species) %>% 
  left_join(species_img, by = "species") %>% 
  pubs_per_cat(category = "species", isotype = TRUE, img_unit = 1)
```

I debated for a while which image to use for humans. In order to avoid racist and sexist connotations, I picked the skull as non-experts can't tell the race and sex from the skull. If you think it's bad to use a skull, then I'll use a Black, Hispanic, Middle Eastern, North African, Central Asian, or South Asian woman's portrait. I stand with the oppressed.

## By journal
```{r}
sort(table(publications$journal))
```

## Location
### General
Just some barplots for number of publications per institution, city, and country. I think city is more informative than country, since obviously, for instance, San Francisco is very different from Anchorage, though they are both in the US. Just within the US, there are regions that are very innovative, and regions that are more repressive. Well, even LA has very different neighborhoods... I just think it's not fair to judge people by whether there's a strong country behind them. First of all, different regions within a countries are not the same. Second, it's not fair to consider someone dumb just because their country is poor and weak; all else being equal, I would admire the person from the poor country more since they had to work harder to get here. I don't care which country is behind you. Right is right, wrong is wrong. Anyway, this is not the place to ramble about politics.
```{r}
pubs_per_cat(publications, "country")
```

How about per capita? Actually this probably does not say much since many researchers in the West are immigrants. Then it raises the burning political question of who counts as the "population".

```{r}
pubs_per_capita(publications, plot = "bar")
```

How about country over time? There might not be enough publications to show a trend.
```{r, fig.height=4, fig.width=4}
pubs_per_year(publications, facet_by = "country")
```
Now look at cities. 

```{r}
pubs_per_cat(publications, "city")
```

Institutions
```{r}
pubs_per_cat(publications, "institution")
```

OK, now here comes the maps!

### Worldwide
```{r}
pubs_on_map2(publications)
```

Let me also plot the per capita thing on a map, as a choropleth
```{r}
pubs_per_capita(publications)
```

Break down by species
```{r, fig.width=5, fig.height=3}
pubs_on_map2(species, facet_by = "species")
```


Probably this does not tell us much since there's only 1 study for some species. Again, we see the received wisdom that research is mostly confined to the West, especially Western Europe, west coast of the US, and New England, but not as much in the other parts of the "West".

By method
```{r, fig.width=5, fig.height=5}
pubs_on_map2(methods, facet_by = "method")
```

I suppose this merely tells us where those techniques were invented and how they spread. It seems that seqFISH and its variants did not spread beyond Caltech; the dataset labeled Harvard is from a collaboration between Long Cai and someone at Harvard. However, MERFISH did spread beyond Harvard, where it originated.

### Europe
Plot a map just for Europe. A problem here is that if I simply filter the `world` sf object, I'll get some islands away from what we usually think is Europe. Kind of feel bad for Russians since Russia is so large that it makes the map look bad and I don't have a paper from Russia in this spreadsheet. 

```{r}
pubs_on_map2(publications, zoom = "europe")
```

```{r}
pubs_per_capita(publications, "europe")
```

I didn't realize that Switzerland and Netherlands were pretty great. Again, this is a small sample size, so I take this with a grain of salt until I plot the other sheets. From the other sheets, I already know that Sweden, the UK, and Germany have many contributions. 

```{r, fig.width=4, fig.height=2.5}
pubs_on_map2(species, zoom = "europe", facet_by = "species")
```

Maybe I just made this plot for fun. Not sure what to say about it.

```{r, fig.width=5, fig.height=4}
pubs_on_map2(methods, zoom = "europe", facet_by = "method", ncol = 2)
```

### USA
Also a plot just for America. I did not encounter a publication in this spreadsheet that is from Hawaii or Alaska, but I'll include Hawaii and Alaska in the map anyway in case I get one in the future.

```{r}
pubs_on_map2(publications, zoom = "usa")
```

```{r}
pubs_per_capita(publications, "usa")
```

```{r, fig.width=5, fig.height=3}
pubs_on_map2(species, zoom = "usa", facet_by = "species")
```

```{r, fig.width=5, fig.height=4}
pubs_on_map2(methods, zoom = "usa", facet_by = "method")
```

# Number of datasets
Since I did not always read the paper very carefully, take this with a grain of salt. Also, sometimes I'm not sure what counts as a dataset. The authors may test a technique on cell culture with just a few genes before moving on to real tissue, and I'm not sure if this should be included. But for microdissection, that's not an issue. Another thing I'm not sure about is biological and technical replica presented together on the paper. Should each replicate count as a dataset or should all the replica count as one dataset as they're intended to answer the same question and are analyzed together as they they were one?
```{r}
sheet %>% 
  count(title, method) %>% 
  ggplot(aes(n)) +
  geom_bar() +
  scale_y_continuous(expand = expansion(c(0, 0.05))) +
  scale_x_continuous(breaks = breaks_width(1))
```

# Word cloud
## Titles
```{r}
plot_wordcloud(sheet)
```

```{r}
plot_wordcloud(sheet, species_use = "Mus musculus")
```

```{r}
plot_wordcloud(sheet, species_use = "Homo sapiens")
```

## Tissues
```{r}
plot_wordcloud(sheet, col_use = "tissue")
```

That 2 comes from U-2 OS
```{r}
plot_wordcloud(sheet, species_use = "Mus musculus", col_use = "tissue")
```

```{r}
plot_wordcloud(sheet, species_use = "Homo sapiens", col_use = "tissue")
```

## Over time

```{r}
range(sheet$date_published)
```

```{r}
plot_wordcloud(sheet, year_min = 2012, year_max = 2016)
```

```{r}
plot_wordcloud(sheet, year_min = 2016, year_max = 2021)
```

## Department names
```{r}
institution_words <- readRDS("output/inst_words.rds")
```

```{r}
plot_wordcloud(sheet, col_use = "department", other_stop_words = institution_words)
```

## Downstream analyses
```{r}
plot_wordcloud(sheet, "downstream")
```

# Programming languages
```{r}
langs <- sheet %>% 
  select(year, title, language) %>% 
  filter(!is.na(language)) %>% 
  distinct() %>% 
  mutate(language = str_split(language, "; ")) %>% 
  unnest(cols = "language")
```

```{r}
lang_img <- readRDS("output/lang_img.rds")
lang_img$image <- purrr::map(lang_img$image_paths, image_read)
```

```{r}
langs %>% 
  inner_join(lang_img, by = "language") %>% 
  pubs_per_cat(category = "language", isotype = TRUE, img_unit = 2)
```

# Data and code availability

How many publications have provided a code repo?
```{r}
ggplot(publications, aes(year, color = !is.na(repo))) +
  geom_freqpoly(binwidth = 1)
```

Here one paper can have multiple datasets and the availability status of the datasets can be different, so I'm plotting the datasets rather than publications here.

```{r}
ggplot(sheet, aes(year, color = has_matrix)) +
  geom_freqpoly(binwidth = 1)
```

It seems that more recent publications are more likely to provide a repo, but not data. Is this significant? Here I fit a linear model to use year to predict proportion of publications with repo or accession and test if the coefficients are 0.
```{r}
pub_repos <- publications %>% 
  group_by(year) %>% 
  summarize(prop_repo = sum(!is.na(repo))/length(repo))
summary(lm(prop_repo ~ year, data = pub_repos))
```

So there sort of is a positive association between year and whether repo is available, and more recent publications are more likely to have repo.

# Number of genes and cells
```{r}
sheet %>% 
  filter(!is.na(n_genes), !is.na(n_cells)) %>% 
  ggplot(aes(date_published, n_genes, color = method)) +
  geom_jitter()
```

While the max possible number of genes increased, I don't think people necessarily want to go for the max.
```{r}
sheet %>% 
  filter(!is.na(n_genes), !is.na(n_cells), n_genes < 2000) %>% 
  ggplot(aes(date_published, n_genes, color = method)) +
  geom_jitter()
```

```{r}
sheet %>% 
  filter(!is.na(n_genes), !is.na(n_cells)) %>% 
  ggplot(aes(date_published, n_cells, color = method)) +
  geom_jitter()
```

```{r}
sheet %>% 
  filter(!is.na(n_genes), !is.na(n_cells), n_cells < 1.5e5) %>% 
  ggplot(aes(date_published, n_cells, color = method)) +
  geom_jitter()
```

I don't think there really is a trend either.
