<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Prequel • museumst</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Prequel">
<meta property="og:description" content="museumst">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">museumst</a>
        <span class="version label label-danger" data-toggle="tooltip" data-placement="bottom" title="Unreleased version">0.0.0.9002</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/analysis.html">Analysis</a>
    </li>
    <li>
      <a href="../articles/array.html">Array</a>
    </li>
    <li>
      <a href="../articles/comparison.html">Comparisons between sheets</a>
    </li>
    <li>
      <a href="../articles/iss.html">ISS</a>
    </li>
    <li>
      <a href="../articles/microdissection.html">Microdissection</a>
    </li>
    <li>
      <a href="../articles/prequel2.html">Prequel</a>
    </li>
    <li>
      <a href="../articles/prequel_analysis.html">Prequel Analysis</a>
    </li>
    <li>
      <a href="../articles/smfish.html">smFISH</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="prequel2_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Prequel</h1>
                        <h4 class="author">Lambda Moses</h4>
            
            <h4 class="date">2020-07-29</h4>
      
      
      <div class="hidden name"><code>prequel2.Rmd</code></div>

    </div>

    
    
<p>Here I explore the history and geography of spatial transcriptomics in the prequel era through the metadata I collected, namely things like in situ atlases.</p>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw">museumst</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="http://ggplot2.tidyverse.org">ggplot2</a></span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="http://purrr.tidyverse.org">purrr</a></span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="http://stringr.tidyverse.org">stringr</a></span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="https://dplyr.tidyverse.org">dplyr</a></span>)
</pre></div>
<pre><code>## 
## Attaching package: 'dplyr'</code></pre>
<pre><code>## The following objects are masked from 'package:stats':
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     intersect, setdiff, setequal, union</code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="kw"><a href="https://r-spatial.github.io/sf">sf</a></span>)
</pre></div>
<pre><code>## Linking to GEOS 3.8.1, GDAL 3.1.1, PROJ 6.3.1</code></pre>
<div class="sourceCode" id="cb7"><pre class="downlit">
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme_get.html">theme_set</a></span>(<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span>())
</pre></div>
<div id="import-data" class="section level1">
<h1 class="hasAnchor">
<a href="#import-data" class="anchor"></a>Import data</h1>
<p>By default, the <code>read_metadata</code> function reads the appropriate sheet from a cache within this package. This may be outdated, since the Google Sheets are constantly updated. Use the argument <code>update = TRUE</code> to download the most up to date version. The default is <code>FALSE</code> to avoid API limits during automated CRAN checks.</p>
<div class="sourceCode" id="cb8"><pre class="downlit">
<span class="kw">sheet</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_metadata.html">read_metadata</a></span>(<span class="st">"Prequel"</span>)
</pre></div>
</div>
<div id="number-of-publications" class="section level1">
<h1 class="hasAnchor">
<a href="#number-of-publications" class="anchor"></a>Number of publications</h1>
<div class="sourceCode" id="cb9"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/duplicated.html">anyDuplicated</a></span>(<span class="kw">sheet</span><span class="op">$</span><span class="kw">title</span>[<span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="kw">sheet</span><span class="op">$</span><span class="kw">title</span>)])
</pre></div>
<pre><code>## [1] 0</code></pre>
<p>How many publications are there in total in this sheet?</p>
<div class="sourceCode" id="cb11"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="kw">sheet</span>)
</pre></div>
<pre><code>## [1] 195</code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit">
<span class="co"># For maps later in this notebook</span>
<span class="kw">city_gc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/geocode_inst_city.html">geocode_inst_city</a></span>(<span class="kw">sheet</span>)
<span class="kw">pubs_on_map2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span>(<span class="kw">pubs_on_map</span>, city_gc = <span class="kw">city_gc</span>)
</pre></div>
<div id="overall" class="section level2">
<h2 class="hasAnchor">
<a href="#overall" class="anchor"></a>Overall</h2>
<div class="sourceCode" id="cb14"><pre class="downlit">
<span class="kw">events</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_major_events.html">read_major_events</a></span>()
</pre></div>
<div class="sourceCode" id="cb15"><pre class="downlit">
<span class="kw">events</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="kw">category</span> <span class="op">==</span> <span class="st">"trap"</span>) <span class="op">%&gt;%</span> 
  <span class="fu"><a href="../reference/plot_timeline.html">plot_timeline</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>, <span class="op">-</span><span class="fl">1</span>), length.out = <span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="kw">.</span>)), 
                expand_x = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0.1</span>, <span class="fl">0.1</span>), expand_y = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0.05</span>, <span class="fl">0.05</span>))
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-7-1.png" width="768"></p>
<div class="sourceCode" id="cb16"><pre class="downlit">
<span class="kw">events</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="kw">category</span> <span class="op">==</span> <span class="st">"ISH atlas"</span>) <span class="op">%&gt;%</span> 
  <span class="fu"><a href="../reference/plot_timeline.html">plot_timeline</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0.3</span>, <span class="op">-</span><span class="fl">0.3</span>, <span class="fl">0.3</span>, <span class="op">-</span><span class="fl">0.3</span>, <span class="fl">0.3</span>, <span class="op">-</span><span class="fl">0.6</span>, <span class="fl">0.3</span>, <span class="op">-</span><span class="fl">0.5</span>, <span class="fl">0.6</span>, <span class="op">-</span><span class="fl">0.7</span>, <span class="fl">0.3</span>, <span class="op">-</span><span class="fl">0.2</span>, <span class="fl">0.6</span>, <span class="op">-</span><span class="fl">0.5</span>, <span class="fl">0.2</span>), 
                expand_x = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0.1</span>, <span class="fl">0.1</span>), expand_y = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0.05</span>, <span class="fl">0.05</span>))
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-8-1.png" width="768"></p>
<p>By default, preprints are excluded from plots of number of publications over time, since the timing of preprints is incoherent to that of published papers. Including preprints will inflate the number of publications in recent months, and if we make the same plot a few months later, that inflation we see now will be gone and moved to a more recent date. You can add the <code>preprints = TRUE</code> argument to all functions plotting things over time in this package to include preprints. You can also use the <code>binwidth</code> argument to specify bin width in days. The default is 365 days.</p>
<div class="sourceCode" id="cb17"><pre class="downlit">
<span class="fu"><a href="../reference/pubs_per_year.html">pubs_per_year</a></span>(<span class="kw">sheet</span>)
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-9-1.png" width="700"></p>
</div>
<div id="by-method" class="section level2">
<h2 class="hasAnchor">
<a href="#by-method" class="anchor"></a>By method</h2>
<div class="sourceCode" id="cb18"><pre class="downlit">
<span class="kw">methods</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/unnest_cat.html">unnest_cat</a></span>(<span class="kw">sheet</span>, <span class="kw">method</span>)
</pre></div>
<div class="sourceCode" id="cb19"><pre class="downlit">
<span class="fu"><a href="../reference/pubs_per_cat.html">pubs_per_cat</a></span>(<span class="kw">methods</span>, <span class="kw">method</span>)
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-11-1.png" width="700"></p>
<div class="sourceCode" id="cb20"><pre class="downlit">
<span class="fu"><a href="../reference/pubs_per_year.html">pubs_per_year</a></span>(<span class="kw">methods</span>, facet_by = <span class="st">"method"</span>)
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-12-1.png" width="576"></p>
</div>
<div id="by-species" class="section level2">
<h2 class="hasAnchor">
<a href="#by-species" class="anchor"></a>By species</h2>
<div class="sourceCode" id="cb21"><pre class="downlit">
<span class="kw">species</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/unnest_cat.html">unnest_cat</a></span>(<span class="kw">sheet</span>, <span class="kw">species</span>)
</pre></div>
<div class="sourceCode" id="cb22"><pre class="downlit">
<span class="fu"><a href="../reference/pubs_per_cat.html">pubs_per_cat</a></span>(<span class="kw">species</span>, <span class="kw">species</span>)
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
<div class="sourceCode" id="cb23"><pre class="downlit">
<span class="fu"><a href="../reference/pubs_per_cat.html">pubs_per_cat</a></span>(<span class="kw">species</span>, <span class="kw">species</span>, n_top = <span class="fl">5</span>, isotype = <span class="fl">TRUE</span>, img_unit = <span class="fl">10</span>)
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
</div>
<div id="by-journal" class="section level2">
<h2 class="hasAnchor">
<a href="#by-journal" class="anchor"></a>By journal</h2>
<div class="sourceCode" id="cb24"><pre class="downlit">
<span class="fu"><a href="../reference/cat_heatmap.html">cat_heatmap</a></span>(<span class="kw">methods</span>, <span class="kw">journal</span>, <span class="kw">method</span>)
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-16-1.png" width="700"></p>
<div class="sourceCode" id="cb25"><pre class="downlit">
<span class="fu"><a href="../reference/cat_heatmap.html">cat_heatmap</a></span>(<span class="kw">species</span>, <span class="kw">journal</span>, <span class="kw">species</span>)
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-17-1.png" width="700"></p>
</div>
<div id="location" class="section level2">
<h2 class="hasAnchor">
<a href="#location" class="anchor"></a>Location</h2>
<div id="general" class="section level3">
<h3 class="hasAnchor">
<a href="#general" class="anchor"></a>General</h3>
<p>Just some barplots for number of publications per institution, city, and country.</p>
<div class="sourceCode" id="cb26"><pre class="downlit">
<span class="fu"><a href="../reference/pubs_per_cat.html">pubs_per_cat</a></span>(<span class="kw">sheet</span>, <span class="kw">country</span>)
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-18-1.png" width="700"></p>
<p>How about per capita? Actually this probably does not say much since many researchers in the West are immigrants. Then it raises the burning political question of who counts as the “population”.</p>
<div class="sourceCode" id="cb27"><pre class="downlit">
<span class="fu"><a href="../reference/pubs_per_capita.html">pubs_per_capita</a></span>(<span class="kw">sheet</span>, plot = <span class="st">"bar"</span>)
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-19-1.png" width="700"></p>
<p>How about country over time? There might not be enough publications to show a trend.</p>
<div class="sourceCode" id="cb28"><pre class="downlit">
<span class="fu"><a href="../reference/pubs_per_year.html">pubs_per_year</a></span>(<span class="kw">sheet</span>, facet_by = <span class="st">"country"</span>)
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-20-1.png" width="576"> Now look at cities.</p>
<div class="sourceCode" id="cb29"><pre class="downlit">
<span class="fu"><a href="../reference/pubs_per_cat.html">pubs_per_cat</a></span>(<span class="kw">sheet</span>, <span class="kw">city</span>)
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-21-1.png" width="576"></p>
<p>Bar Harbor is for GXD(JAX), and Edinburgh is for EMAGE Institutions</p>
<div class="sourceCode" id="cb30"><pre class="downlit">
<span class="fu"><a href="../reference/pubs_per_cat.html">pubs_per_cat</a></span>(<span class="kw">sheet</span>, <span class="kw">institution</span>)
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-22-1.png" width="576"></p>
</div>
<div id="worldwide" class="section level3">
<h3 class="hasAnchor">
<a href="#worldwide" class="anchor"></a>Worldwide</h3>
<div class="sourceCode" id="cb31"><pre class="downlit">
<span class="fu">pubs_on_map2</span>(<span class="kw">sheet</span>)
</pre></div>
<pre><code>## Warning: Removed 2 rows containing missing values (geom_label_repel).</code></pre>
<p><img src="prequel2_files/figure-html/unnamed-chunk-23-1.png" width="700"></p>
<p>Still, it’s mostly in the US and Western Europe, but this time we see some more in Asia. Still no Africa and South America.</p>
<p>Let me also plot the per capita thing on a map, as a choropleth</p>
<div class="sourceCode" id="cb33"><pre class="downlit">
<span class="fu"><a href="../reference/pubs_per_capita.html">pubs_per_capita</a></span>(<span class="kw">sheet</span>)
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-24-1.png" width="700"></p>
<p>Break down by species</p>
<div class="sourceCode" id="cb34"><pre class="downlit">
<span class="fu">pubs_on_map2</span>(<span class="kw">species</span>, facet_by = <span class="st">"species"</span>, ncol = <span class="fl">4</span>)
</pre></div>
<pre><code>## Warning: Removed 4 rows containing missing values (geom_label_repel).</code></pre>
<p><img src="prequel2_files/figure-html/unnamed-chunk-25-1.png" width="768"></p>
<p>Ciona is more centered in Japan, and chicken more in Arizona. I already know that.</p>
<p>Probably this does not tell us much since there’s only 1 study for some species. Again, we see the received wisdom that research is mostly confined to the West, especially Western Europe, west coast of the US, and New England, but not as much in the other parts of the “West”.</p>
</div>
<div id="europe" class="section level3">
<h3 class="hasAnchor">
<a href="#europe" class="anchor"></a>Europe</h3>
<p>Plot a map just for Europe. A problem here is that if I simply filter the <code>world</code> sf object, I’ll get some islands away from what we usually think is Europe.</p>
<div class="sourceCode" id="cb36"><pre class="downlit">
<span class="fu">pubs_on_map2</span>(<span class="kw">sheet</span>, zoom = <span class="st">"europe"</span>)
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-26-1.png" width="700"></p>
<div class="sourceCode" id="cb37"><pre class="downlit">
<span class="fu"><a href="../reference/pubs_per_capita.html">pubs_per_capita</a></span>(<span class="kw">sheet</span>, <span class="st">"europe"</span>)
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-27-1.png" width="700"></p>
<p>I’m kind of amazed that Sweden didn’t contribute to the prequel stuff, yet it’s doing so well in the newer spatial transcriptomics technologies.</p>
<div class="sourceCode" id="cb38"><pre class="downlit">
<span class="fu">pubs_on_map2</span>(<span class="kw">species</span>, zoom = <span class="st">"europe"</span>, facet_by = <span class="st">"species"</span>, ncol = <span class="fl">5</span>)
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-28-1.png" width="528"></p>
<p>Looks like some French labs are quite interested in unconventional model organisms. That’s consistent with our collaboration with that French lab on the jellyfish Clytia project. I still wonder why. Is it a French thing?</p>
</div>
<div id="usa" class="section level3">
<h3 class="hasAnchor">
<a href="#usa" class="anchor"></a>USA</h3>
<p>Also a plot just for America. I did not encounter a publication in this spreadsheet that is from Hawaii or Alaska, but I’ll include Hawaii and Alaska in the map anyway in case I get one in the future.</p>
<div class="sourceCode" id="cb39"><pre class="downlit">
<span class="fu">pubs_on_map2</span>(<span class="kw">sheet</span>, zoom = <span class="st">"usa"</span>)
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-29-1.png" width="700"></p>
<div class="sourceCode" id="cb40"><pre class="downlit">
<span class="fu"><a href="../reference/pubs_per_capita.html">pubs_per_capita</a></span>(<span class="kw">sheet</span>, <span class="st">"usa"</span>)
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-30-1.png" width="700"></p>
<div class="sourceCode" id="cb41"><pre class="downlit">
<span class="fu">pubs_on_map2</span>(<span class="kw">species</span>, zoom = <span class="st">"usa"</span>, facet_by = <span class="st">"species"</span>, ncol = <span class="fl">3</span>)
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-31-1.png" width="768"></p>
<p>ZFIN is based in Eugene, Oregon. GEISHA is based in Tempe, Arizona. GXD is based in JAX, Bar Harbor, Maine. There’re a number of Drosophila projects from Berkeley. And we see that on this map. I still wonder why the prequel stuff is more spread out around the world than the current era stuff. Is that because ISH and WMISH, included the robotic versions, are much cheaper than say ST, Visium, ISS, MERISH, and seqFISH? But how about LCM, which is also quite spread out? LCM certainly requirely expensive specialized equipments. But back in UCLA, I did hear that the histology core did LCM, which made it easier and cheaper for labs that don’t have the equipment, while at present, I don’t think other current era spatial transcriptomics technology is supported by a core facility.</p>
</div>
</div>
</div>
<div id="word-cloud" class="section level1">
<h1 class="hasAnchor">
<a href="#word-cloud" class="anchor"></a>Word cloud</h1>
<div id="titles" class="section level2">
<h2 class="hasAnchor">
<a href="#titles" class="anchor"></a>Titles</h2>
<div class="sourceCode" id="cb42"><pre class="downlit">
<span class="fu"><a href="../reference/plot_wordcloud.html">plot_wordcloud</a></span>(<span class="kw">sheet</span>)
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-32-1.png" width="576"></p>
</div>
<div id="tissues" class="section level2">
<h2 class="hasAnchor">
<a href="#tissues" class="anchor"></a>Tissues</h2>
<div class="sourceCode" id="cb43"><pre class="downlit">
<span class="fu"><a href="../reference/plot_wordcloud.html">plot_wordcloud</a></span>(<span class="kw">sheet</span>, <span class="kw">tissue</span>)
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-33-1.png" width="700"></p>
<div class="sourceCode" id="cb44"><pre class="downlit">
<span class="fu"><a href="../reference/plot_wordcloud.html">plot_wordcloud</a></span>(<span class="kw">sheet</span>, <span class="kw">tissue</span>, species_use = <span class="st">"Drosophila melanogaster"</span>)
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-34-1.png" width="576"></p>
<div class="sourceCode" id="cb45"><pre class="downlit">
<span class="fu"><a href="../reference/plot_wordcloud.html">plot_wordcloud</a></span>(<span class="kw">sheet</span>, <span class="kw">tissue</span>, species_use = <span class="st">"Mus musculus"</span>)
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-35-1.png" width="700"></p>
</div>
<div id="over-time" class="section level2">
<h2 class="hasAnchor">
<a href="#over-time" class="anchor"></a>Over time</h2>
<div class="sourceCode" id="cb46"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/range.html">range</a></span>(<span class="kw">sheet</span><span class="op">$</span><span class="kw">date_published</span>)
</pre></div>
<pre><code>## [1] "1987-12-01" "2020-02-09"</code></pre>
<div class="sourceCode" id="cb48"><pre class="downlit">
<span class="fu"><a href="../reference/plot_wordcloud.html">plot_wordcloud</a></span>(<span class="kw">sheet</span>, year_min = <span class="fl">1987</span>, year_max = <span class="fl">1995</span>, scale = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">4</span>, <span class="fl">0.1</span>))
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-37-1.png" width="672"></p>
<div class="sourceCode" id="cb49"><pre class="downlit">
<span class="fu"><a href="../reference/plot_wordcloud.html">plot_wordcloud</a></span>(<span class="kw">sheet</span>, year_min = <span class="fl">1995</span>, year_max = <span class="fl">2000</span>)
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-38-1.png" width="576"></p>
<div class="sourceCode" id="cb50"><pre class="downlit">
<span class="fu"><a href="../reference/plot_wordcloud.html">plot_wordcloud</a></span>(<span class="kw">sheet</span>, year_min = <span class="fl">2000</span>, year_max = <span class="fl">2005</span>)
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-39-1.png" width="576"></p>
<div class="sourceCode" id="cb51"><pre class="downlit">
<span class="fu"><a href="../reference/plot_wordcloud.html">plot_wordcloud</a></span>(<span class="kw">sheet</span>, year_min = <span class="fl">2005</span>, year_max = <span class="fl">2010</span>)
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-40-1.png" width="576"></p>
<div class="sourceCode" id="cb52"><pre class="downlit">
<span class="fu"><a href="../reference/plot_wordcloud.html">plot_wordcloud</a></span>(<span class="kw">sheet</span>, year_min = <span class="fl">2010</span>, year_max = <span class="fl">2021</span>, scale = <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">5</span>, <span class="fl">0.1</span>))
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-41-1.png" width="576"></p>
</div>
<div id="department-names" class="section level2">
<h2 class="hasAnchor">
<a href="#department-names" class="anchor"></a>Department names</h2>
<div class="sourceCode" id="cb53"><pre class="downlit">
<span class="fu"><a href="../reference/plot_wordcloud.html">plot_wordcloud</a></span>(<span class="kw">sheet</span>, col_use = <span class="st">"department"</span>, other_stop_words = <span class="kw">inst_words</span>)
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-42-1.png" width="576"></p>
</div>
</div>
<div id="data-availability" class="section level1">
<h1 class="hasAnchor">
<a href="#data-availability" class="anchor"></a>Data availability</h1>
<p>How many of the resources are still available?</p>
<div class="sourceCode" id="cb54"><pre class="downlit">
<span class="fu"><a href="../reference/hist_bool.html">hist_bool</a></span>(<span class="kw">sheet</span>, <span class="kw">still_available</span>)
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-43-1.png" width="700"></p>
<p>It seems that more recent ones are more likely to be still available. The NAs are for those atlases that has never been made available. Well, actually that might not be the case, since the data from those papers has most likely now been deposited in one of the databases like GXD and ZFIN. It seems that the NAs tend to be older.</p>
<div class="sourceCode" id="cb55"><pre class="downlit">
<span class="fu"><a href="../reference/test_year_bool.html">test_year_bool</a></span>(<span class="kw">sheet</span>, <span class="kw">still_available</span>)
</pre></div>
<pre><code>## 
## Call:
## glm(formula = bool_use ~ date_published, family = "binomial", 
##     data = pubs)
## 
## Deviance Residuals: 
##     Min       1Q   Median       3Q      Max  
## -2.2206   0.2444   0.3909   0.6538   1.2988  
## 
## Coefficients:
##                  Estimate Std. Error z value Pr(&gt;|z|)    
## (Intercept)    -5.4685494  1.8700349  -2.924 0.003452 ** 
## date_published  0.0005119  0.0001417   3.612 0.000304 ***
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
## 
## (Dispersion parameter for binomial family taken to be 1)
## 
##     Null deviance: 105.105  on 111  degrees of freedom
## Residual deviance:  88.468  on 110  degrees of freedom
##   (83 observations deleted due to missingness)
## AIC: 92.468
## 
## Number of Fisher Scoring iterations: 5</code></pre>
<p>Yes, it is significant. More recent databases are more likely to be still available.</p>
</div>
<div id="number-of-genes" class="section level1">
<h1 class="hasAnchor">
<a href="#number-of-genes" class="anchor"></a>Number of genes</h1>
<p>For the old enhancer and gene trap studies, there weren’t genome annotations and number of lines was reported rather than genes. For old ISH atlases, the number of clones was reported.</p>
<div class="sourceCode" id="cb57"><pre class="downlit">
<span class="fu"><a href="../reference/pubs_per_year.html">pubs_per_year</a></span>(<span class="kw">sheet</span>, facet_by = <span class="st">"item_type"</span>, n_top = <span class="fl">3</span>)
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-45-1.png" width="700"></p>
<p>Yeah, genes became prevalent later, and the prevalence of clones is quite transient. However, lines are not dead, since enhancer and gene traps are not dead since those are high throughput perturbations in addition to visualization.</p>
<p>Do more recent studies tend to do more genes?</p>
<div class="sourceCode" id="cb58"><pre class="downlit">
<span class="kw">sheet</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="op">!</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_detect.html">str_detect</a></span>(<span class="kw">method</span>, <span class="st">"collection"</span>), <span class="kw">item_type</span> <span class="op">==</span> <span class="st">"gene"</span>) <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">date_published</span>, <span class="kw">n_items</span>)) <span class="op">+</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span>()
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-46-1.png" width="700"></p>
<p>Not really.</p>
<p>I know that studies that do a particular type of genes tend to have fewer genes. Which types of genes?</p>
<div class="sourceCode" id="cb59"><pre class="downlit">
<span class="kw">sheet</span> <span class="op">%&gt;%</span> 
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="kw">gene_type</span>)) <span class="op">%&gt;%</span> 
  <span class="fu"><a href="../reference/pubs_per_cat.html">pubs_per_cat</a></span>(<span class="kw">gene_type</span>)
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-47-1.png" width="700"></p>
<p>Are more recent publications more likely to focus on some type of genes?</p>
<div class="sourceCode" id="cb60"><pre class="downlit">
<span class="fu"><a href="../reference/hist_bool.html">hist_bool</a></span>(<span class="kw">sheet</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="kw">gene_type</span>))
</pre></div>
<p><img src="prequel2_files/figure-html/unnamed-chunk-48-1.png" width="700"></p>
<div class="sourceCode" id="cb61"><pre class="downlit">
<span class="fu"><a href="../reference/test_year_bool.html">test_year_bool</a></span>(<span class="kw">sheet</span>, <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="kw">gene_type</span>))
</pre></div>
<pre><code>## 
## Call:
## glm(formula = bool_use ~ date_published, family = "binomial", 
##     data = pubs)
## 
## Deviance Residuals: 
##     Min       1Q   Median       3Q      Max  
## -0.5678  -0.5367  -0.5176  -0.5018   2.0903  
## 
## Coefficients:
##                  Estimate Std. Error z value Pr(&gt;|z|)  
## (Intercept)    -2.399e+00  1.174e+00  -2.043    0.041 *
## date_published  3.582e-05  8.525e-05   0.420    0.674  
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
## 
## (Dispersion parameter for binomial family taken to be 1)
## 
##     Null deviance: 149.35  on 194  degrees of freedom
## Residual deviance: 149.18  on 193  degrees of freedom
## AIC: 153.18
## 
## Number of Fisher Scoring iterations: 4</code></pre>
<p>I don’t think so.</p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Lambda Moses, Lior Pachter.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.9000.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
