<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>smFISH • museumst</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/minty/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="smFISH">
<meta property="og:description" content="museumst">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">museumst</a>
        <span class="version label label-danger" data-toggle="tooltip" data-placement="bottom" title="Unreleased version">0.0.0.9002</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/analysis.html">Analysis</a>
    </li>
    <li>
      <a href="../articles/array.html">Array</a>
    </li>
    <li>
      <a href="../articles/comparison.html">Comparisons between sheets</a>
    </li>
    <li>
      <a href="../articles/iss.html">ISS</a>
    </li>
    <li>
      <a href="../articles/microdissection.html">Microdissection</a>
    </li>
    <li>
      <a href="../articles/prequel2.html">Prequel</a>
    </li>
    <li>
      <a href="../articles/prequel_analysis.html">Prequel Analysis</a>
    </li>
    <li>
      <a href="../articles/smfish.html">smFISH</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="smfish_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>smFISH</h1>
                        <h4 class="author">Lambda Moses</h4>
            
            <h4 class="date">2020-07-29</h4>
      
      
      <div class="hidden name"><code>smfish.Rmd</code></div>

    </div>

    
    
<p>Here I explore the history and geography of spatial transcriptomics based on smFISH through the metadata I collected.</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">museumst</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">ggplot2</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">purrr</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">dplyr</span>)</pre></body></html></div>
<pre><code>## 
## Attaching package: 'dplyr'</code></pre>
<pre><code>## The following objects are masked from 'package:stats':
## 
##     filter, lag</code></pre>
<pre><code>## The following objects are masked from 'package:base':
## 
##     intersect, setdiff, setequal, union</code></pre>
<div class="sourceCode" id="cb5"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">sf</span>)</pre></body></html></div>
<pre><code>## Warning: package 'sf' was built under R version 4.0.2</code></pre>
<pre><code>## Linking to GEOS 3.8.1, GDAL 3.1.1, PROJ 6.3.1</code></pre>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">ggrepel</span>)
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme_get.html">theme_set</a></span>(<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span>())</pre></body></html></div>
<div id="import-data" class="section level1">
<h1 class="hasAnchor">
<a href="#import-data" class="anchor"></a>Import data</h1>
<p>By default, the <code>read_metadata</code> function reads the appropriate sheet from a cache within this package. This may be outdated, since the Google Sheets are constantly updated. Use the argument <code>update = TRUE</code> to download the most up to date version. The default is <code>FALSE</code> to avoid API limits during automated CRAN checks.</p>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="no">sheet</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/read_metadata.html">read_metadata</a></span>(<span class="st">"smFISH"</span>)</pre></body></html></div>
</div>
<div id="number-of-publications" class="section level1">
<h1 class="hasAnchor">
<a href="#number-of-publications" class="anchor"></a>Number of publications</h1>
<p>How many publications per year?</p>
<div class="sourceCode" id="cb10"><html><body><pre class="r"><span class="co"># Remove duplicates as different datasets in the same publication have their own rows</span>
<span class="no">publications</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/get_pubs_df.html">get_pubs_df</a></span>(<span class="no">sheet</span>, <span class="st">"repo"</span>)</pre></body></html></div>
<p>How many publications are there in total in this sheet?</p>
<div class="sourceCode" id="cb11"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="no">publications</span>)</pre></body></html></div>
<pre><code>## [1] 30</code></pre>
<div class="sourceCode" id="cb13"><html><body><pre class="r"><span class="co"># For maps later in this notebook</span>
<span class="no">city_gc</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/geocode_inst_city.html">geocode_inst_city</a></span>(<span class="no">publications</span>)
<span class="no">pubs_on_map2</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://purrr.tidyverse.org/reference/partial.html">partial</a></span>(<span class="no">pubs_on_map</span>, <span class="kw">city_gc</span> <span class="kw">=</span> <span class="no">city_gc</span>)</pre></body></html></div>
<div id="overall" class="section level2">
<h2 class="hasAnchor">
<a href="#overall" class="anchor"></a>Overall</h2>
<p>Number of publications per year. Preprints are excluded from plots over years as it takes months to publish so the dates of the preprints are incoherent with those for the other papers.</p>
<div class="sourceCode" id="cb14"><html><body><pre class="r"><span class="no">events</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/read_major_events.html">read_major_events</a></span>()</pre></body></html></div>
<div class="sourceCode" id="cb15"><html><body><pre class="r"><span class="no">events</span> <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="no">category</span> <span class="kw">==</span> <span class="st">"smFISH"</span>) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="../reference/plot_timeline.html">plot_timeline</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0.8</span>, -<span class="fl">0.5</span>, <span class="fl">0.6</span>, -<span class="fl">0.5</span>, <span class="fl">1</span>, <span class="fl">0.6</span>, -<span class="fl">0.4</span>, <span class="fl">0.7</span>),
                <span class="kw">expand_x</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0.1</span>, <span class="fl">0.1</span>), <span class="kw">expand_y</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0.05</span>, <span class="fl">0.05</span>))</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-7-1.png" width="768"></p>
<div class="sourceCode" id="cb16"><html><body><pre class="r"><span class="fu"><a href="../reference/pubs_per_year.html">pubs_per_year</a></span>(<span class="no">publications</span>)</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
</div>
<div id="by-method" class="section level2">
<h2 class="hasAnchor">
<a href="#by-method" class="anchor"></a>By method</h2>
<p>How about when broken down by method (not for data analysis sheets)?</p>
<div class="sourceCode" id="cb17"><html><body><pre class="r"><span class="no">methods</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/unnest_cat.html">unnest_cat</a></span>(<span class="no">sheet</span>, <span class="no">method</span>)</pre></body></html></div>
<div class="sourceCode" id="cb18"><html><body><pre class="r"><span class="fu"><a href="../reference/pubs_per_year.html">pubs_per_year</a></span>(<span class="no">methods</span>, <span class="kw">facet_by</span> <span class="kw">=</span> <span class="st">"method"</span>)</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-10-1.png" width="576"></p>
</div>
<div id="by-species" class="section level2">
<h2 class="hasAnchor">
<a href="#by-species" class="anchor"></a>By species</h2>
<div class="sourceCode" id="cb19"><html><body><pre class="r"><span class="no">species</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/unnest_cat.html">unnest_cat</a></span>(<span class="no">sheet</span>, <span class="no">species</span>)</pre></body></html></div>
<div class="sourceCode" id="cb20"><html><body><pre class="r"><span class="fu"><a href="../reference/pubs_per_cat.html">pubs_per_cat</a></span>(<span class="no">species</span>, <span class="no">species</span>)</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-12-1.png" width="700"></p>
<div class="sourceCode" id="cb21"><html><body><pre class="r"><span class="fu"><a href="../reference/pubs_per_cat.html">pubs_per_cat</a></span>(<span class="no">species</span>, <span class="no">species</span>, <span class="kw">isotype</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">img_unit</span> <span class="kw">=</span> <span class="fl">1</span>)</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-13-1.png" width="576"></p>
</div>
<div id="by-journal" class="section level2">
<h2 class="hasAnchor">
<a href="#by-journal" class="anchor"></a>By journal</h2>
<div class="sourceCode" id="cb22"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/sort.html">sort</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="no">publications</span>$<span class="no">journal</span>))</pre></body></html></div>
<pre><code>## 
##                      Cell            Genome Biology            Molecular Cell 
##                         1                         1                         1 
## Molecular Systems Biology      Nature Biotechnology       Nature Neuroscience 
##                         1                         1                         1 
##                    Neuron              Cell Systems                    Nature 
##                         1                         2                         2 
##     Nature Communications                   Science        Scientific Reports 
##                         2                         2                         2 
##                      PNAS                   bioRxiv            Nature Methods 
##                         3                         5                         5</code></pre>
</div>
<div id="location" class="section level2">
<h2 class="hasAnchor">
<a href="#location" class="anchor"></a>Location</h2>
<div id="general" class="section level3">
<h3 class="hasAnchor">
<a href="#general" class="anchor"></a>General</h3>
<p>Just some barplots for number of publications per institution, city, and country.</p>
<div class="sourceCode" id="cb24"><html><body><pre class="r"><span class="fu"><a href="../reference/pubs_per_cat.html">pubs_per_cat</a></span>(<span class="no">publications</span>, <span class="no">country</span>)</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
<p>How about per capita? Actually this probably does not say much since many researchers in the West are immigrants. Then it raises the burning political question of who counts as the “population”.</p>
<div class="sourceCode" id="cb25"><html><body><pre class="r"><span class="fu"><a href="../reference/pubs_per_capita.html">pubs_per_capita</a></span>(<span class="no">publications</span>, <span class="kw">plot</span> <span class="kw">=</span> <span class="st">"bar"</span>)</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-16-1.png" width="700"></p>
<p>How about country over time? There might not be enough publications to show a trend.</p>
<div class="sourceCode" id="cb26"><html><body><pre class="r"><span class="fu"><a href="../reference/pubs_per_year.html">pubs_per_year</a></span>(<span class="no">publications</span>, <span class="kw">facet_by</span> <span class="kw">=</span> <span class="st">"country"</span>)</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-17-1.png" width="576"></p>
<p>Well, I know, this is because seqFISH comes from Caltech and MERFISH comes from Harvard.</p>
<p>Now look at cities.</p>
<div class="sourceCode" id="cb27"><html><body><pre class="r"><span class="fu"><a href="../reference/pubs_per_cat.html">pubs_per_cat</a></span>(<span class="no">publications</span>, <span class="no">city</span>)</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-18-1.png" width="700"></p>
<p>Cambridge for Harvard, and Pasadena for Caltech.</p>
<p>Institutions</p>
<div class="sourceCode" id="cb28"><html><body><pre class="r"><span class="fu"><a href="../reference/pubs_per_cat.html">pubs_per_cat</a></span>(<span class="no">publications</span>, <span class="no">institution</span>)</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-19-1.png" width="700"></p>
<p>So thank you, Long Cai. OK, now here comes the maps!</p>
</div>
<div id="worldwide" class="section level3">
<h3 class="hasAnchor">
<a href="#worldwide" class="anchor"></a>Worldwide</h3>
<div class="sourceCode" id="cb29"><html><body><pre class="r"><span class="fu">pubs_on_map2</span>(<span class="no">publications</span>)</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-20-1.png" width="700"></p>
<p>Let me also plot the per capita thing on a map, as a choropleth</p>
<div class="sourceCode" id="cb30"><html><body><pre class="r"><span class="fu"><a href="../reference/pubs_per_capita.html">pubs_per_capita</a></span>(<span class="no">publications</span>)</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-21-1.png" width="700"></p>
<p>Break down by species</p>
<div class="sourceCode" id="cb31"><html><body><pre class="r"><span class="fu">pubs_on_map2</span>(<span class="no">species</span>, <span class="kw">facet_by</span> <span class="kw">=</span> <span class="st">"species"</span>, <span class="kw">ncol</span> <span class="kw">=</span> <span class="fl">2</span>)</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-22-1.png" width="700"></p>
<p>By method</p>
<div class="sourceCode" id="cb32"><html><body><pre class="r"><span class="fu">pubs_on_map2</span>(<span class="no">methods</span>, <span class="kw">facet_by</span> <span class="kw">=</span> <span class="st">"method"</span>)</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-23-1.png" width="768"></p>
<p>I suppose this merely tells us where those techniques were invented and how they spread. It seems that seqFISH and its variants did not spread beyond Caltech; the dataset labeled Harvard is from a collaboration between Long Cai and someone at Harvard. However, MERFISH did spread beyond Harvard, where it originated, to UCLA.</p>
</div>
<div id="europe" class="section level3">
<h3 class="hasAnchor">
<a href="#europe" class="anchor"></a>Europe</h3>
<p>Plot a map just for Europe. A problem here is that if I simply filter the <code>world</code> sf object, I’ll get some islands away from what we usually think is Europe. Kind of feel bad for Russians since Russia is so large that it makes the map look bad and I don’t have a paper from Russia in this spreadsheet.</p>
<div class="sourceCode" id="cb33"><html><body><pre class="r"><span class="fu">pubs_on_map2</span>(<span class="no">publications</span>, <span class="kw">zoom</span> <span class="kw">=</span> <span class="st">"europe"</span>)</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-24-1.png" width="700"></p>
<div class="sourceCode" id="cb34"><html><body><pre class="r"><span class="fu"><a href="../reference/pubs_per_capita.html">pubs_per_capita</a></span>(<span class="no">publications</span>, <span class="st">"europe"</span>)</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-25-1.png" width="700"></p>
<div class="sourceCode" id="cb35"><html><body><pre class="r"><span class="fu">pubs_on_map2</span>(<span class="no">species</span>, <span class="kw">zoom</span> <span class="kw">=</span> <span class="st">"europe"</span>, <span class="kw">facet_by</span> <span class="kw">=</span> <span class="st">"species"</span>)</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-26-1.png" width="576"></p>
<p>Maybe I just made this plot for fun. Not sure what to say about it.</p>
<div class="sourceCode" id="cb36"><html><body><pre class="r"><span class="fu">pubs_on_map2</span>(<span class="no">methods</span>, <span class="kw">zoom</span> <span class="kw">=</span> <span class="st">"europe"</span>, <span class="kw">facet_by</span> <span class="kw">=</span> <span class="st">"method"</span>, <span class="kw">ncol</span> <span class="kw">=</span> <span class="fl">2</span>)</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-27-1.png" width="576"></p>
<p>Note that HybISS and SCRINSHOT both use RCA, which is used in the original 2013 ISS, which comes from Sweden. Is that why?</p>
</div>
<div id="usa" class="section level3">
<h3 class="hasAnchor">
<a href="#usa" class="anchor"></a>USA</h3>
<p>Also a plot just for America. I did not encounter a publication in this spreadsheet that is from Hawaii or Alaska, but I’ll include Hawaii and Alaska in the map anyway in case I get one in the future.</p>
<div class="sourceCode" id="cb37"><html><body><pre class="r"><span class="fu">pubs_on_map2</span>(<span class="no">publications</span>, <span class="kw">zoom</span> <span class="kw">=</span> <span class="st">"usa"</span>)</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-28-1.png" width="700"></p>
<div class="sourceCode" id="cb38"><html><body><pre class="r"><span class="fu"><a href="../reference/pubs_per_capita.html">pubs_per_capita</a></span>(<span class="no">publications</span>, <span class="st">"usa"</span>)</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-29-1.png" width="700"></p>
<div class="sourceCode" id="cb39"><html><body><pre class="r"><span class="fu">pubs_on_map2</span>(<span class="no">species</span>, <span class="kw">zoom</span> <span class="kw">=</span> <span class="st">"usa"</span>, <span class="kw">facet_by</span> <span class="kw">=</span> <span class="st">"species"</span>, <span class="kw">ncol</span> <span class="kw">=</span> <span class="fl">2</span>)</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-30-1.png" width="700"></p>
<div class="sourceCode" id="cb40"><html><body><pre class="r"><span class="fu">pubs_on_map2</span>(<span class="no">methods</span>, <span class="kw">zoom</span> <span class="kw">=</span> <span class="st">"usa"</span>, <span class="kw">facet_by</span> <span class="kw">=</span> <span class="st">"method"</span>)</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-31-1.png" width="768"></p>
</div>
</div>
</div>
<div id="word-cloud" class="section level1">
<h1 class="hasAnchor">
<a href="#word-cloud" class="anchor"></a>Word cloud</h1>
<div id="titles" class="section level2">
<h2 class="hasAnchor">
<a href="#titles" class="anchor"></a>Titles</h2>
<div class="sourceCode" id="cb41"><html><body><pre class="r"><span class="fu"><a href="../reference/plot_wordcloud.html">plot_wordcloud</a></span>(<span class="no">sheet</span>)</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-32-1.png" width="576"></p>
<p>Yeah, single cell.</p>
<div class="sourceCode" id="cb42"><html><body><pre class="r"><span class="fu"><a href="../reference/plot_wordcloud.html">plot_wordcloud</a></span>(<span class="no">sheet</span>, <span class="kw">species_use</span> <span class="kw">=</span> <span class="st">"Mus musculus"</span>)</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-33-1.png" width="576"></p>
<div class="sourceCode" id="cb43"><html><body><pre class="r"><span class="fu"><a href="../reference/plot_wordcloud.html">plot_wordcloud</a></span>(<span class="no">sheet</span>, <span class="kw">species_use</span> <span class="kw">=</span> <span class="st">"Homo sapiens"</span>)</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-34-1.png" width="576"></p>
</div>
<div id="tissues" class="section level2">
<h2 class="hasAnchor">
<a href="#tissues" class="anchor"></a>Tissues</h2>
<div class="sourceCode" id="cb44"><html><body><pre class="r"><span class="fu"><a href="../reference/plot_wordcloud.html">plot_wordcloud</a></span>(<span class="no">sheet</span>, <span class="no">tissue</span>)</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-35-1.png" width="384"></p>
<p>That 2 comes from U-2 OS, which was used many times to test MERFISH.</p>
<div class="sourceCode" id="cb45"><html><body><pre class="r"><span class="fu"><a href="../reference/plot_wordcloud.html">plot_wordcloud</a></span>(<span class="no">sheet</span>, <span class="no">tissue</span>, <span class="kw">species_use</span> <span class="kw">=</span> <span class="st">"Mus musculus"</span>)</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-36-1.png" width="576"></p>
<div class="sourceCode" id="cb46"><html><body><pre class="r"><span class="fu"><a href="../reference/plot_wordcloud.html">plot_wordcloud</a></span>(<span class="no">sheet</span>, <span class="no">tissue</span>, <span class="kw">species_use</span> <span class="kw">=</span> <span class="st">"Homo sapiens"</span>)</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-37-1.png" width="384"></p>
<p>Those are all about cell culture. I look forward to seeing this used on more real tissues.</p>
</div>
<div id="over-time" class="section level2">
<h2 class="hasAnchor">
<a href="#over-time" class="anchor"></a>Over time</h2>
<div class="sourceCode" id="cb47"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/range.html">range</a></span>(<span class="no">sheet</span>$<span class="no">date_published</span>)</pre></body></html></div>
<pre><code>## [1] "2012-06-03" "2020-07-03"</code></pre>
<div class="sourceCode" id="cb49"><html><body><pre class="r"><span class="fu"><a href="../reference/plot_wordcloud.html">plot_wordcloud</a></span>(<span class="no">sheet</span>, <span class="kw">year_min</span> <span class="kw">=</span> <span class="fl">2012</span>, <span class="kw">year_max</span> <span class="kw">=</span> <span class="fl">2016</span>)</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-39-1.png" width="384"></p>
<div class="sourceCode" id="cb50"><html><body><pre class="r"><span class="fu"><a href="../reference/plot_wordcloud.html">plot_wordcloud</a></span>(<span class="no">sheet</span>, <span class="kw">year_min</span> <span class="kw">=</span> <span class="fl">2016</span>, <span class="kw">year_max</span> <span class="kw">=</span> <span class="fl">2021</span>)</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-40-1.png" width="576"></p>
</div>
<div id="department-names" class="section level2">
<h2 class="hasAnchor">
<a href="#department-names" class="anchor"></a>Department names</h2>
<div class="sourceCode" id="cb51"><html><body><pre class="r"><span class="fu"><a href="../reference/plot_wordcloud.html">plot_wordcloud</a></span>(<span class="no">sheet</span>, <span class="kw">col_use</span> <span class="kw">=</span> <span class="st">"department"</span>, <span class="kw">other_stop_words</span> <span class="kw">=</span> <span class="no">inst_words</span>)</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-41-1.png" width="576"></p>
<p>That’s because Long Cai used to be in CCE, though later he moved to BBE.</p>
</div>
<div id="downstream-analyses" class="section level2">
<h2 class="hasAnchor">
<a href="#downstream-analyses" class="anchor"></a>Downstream analyses</h2>
<div class="sourceCode" id="cb52"><html><body><pre class="r"><span class="fu"><a href="../reference/plot_wordcloud.html">plot_wordcloud</a></span>(<span class="no">sheet</span>, <span class="no">downstream</span>)</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-42-1.png" width="480"></p>
</div>
</div>
<div id="programming-languages" class="section level1">
<h1 class="hasAnchor">
<a href="#programming-languages" class="anchor"></a>Programming languages</h1>
<div class="sourceCode" id="cb53"><html><body><pre class="r"><span class="no">langs</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/unnest_cat.html">unnest_cat</a></span>(<span class="no">sheet</span>, <span class="no">language</span>)</pre></body></html></div>
<div class="sourceCode" id="cb54"><html><body><pre class="r"><span class="fu"><a href="../reference/pubs_per_cat.html">pubs_per_cat</a></span>(<span class="no">langs</span>, <span class="no">language</span>, <span class="kw">isotype</span> <span class="kw">=</span> <span class="fl">TRUE</span>, <span class="kw">img_unit</span> <span class="kw">=</span> <span class="fl">2</span>)</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-44-1.png" width="700"></p>
</div>
<div id="data-and-code-availability" class="section level1">
<h1 class="hasAnchor">
<a href="#data-and-code-availability" class="anchor"></a>Data and code availability</h1>
<p>How many publications have provided a code repo?</p>
<div class="sourceCode" id="cb55"><html><body><pre class="r"><span class="fu"><a href="../reference/hist_bool.html">hist_bool</a></span>(<span class="no">publications</span>, !<span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="no">repo</span>))</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-45-1.png" width="700"></p>
<p>Here one paper can have multiple datasets and the availability status of the datasets can be different, so I’m plotting the datasets rather than publications here.</p>
<div class="sourceCode" id="cb56"><html><body><pre class="r"><span class="fu"><a href="../reference/hist_bool.html">hist_bool</a></span>(<span class="no">sheet</span>, <span class="no">has_matrix</span>)</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-46-1.png" width="700"></p>
<p>It seems that more recent publications are more likely to provide a repo, but not data. Is this significant? Here I fit a logistic regression model to use date published to predict whether publications have repo or accession and test if the coefficients are 0.</p>
<div class="sourceCode" id="cb57"><html><body><pre class="r"><span class="fu"><a href="../reference/test_year_bool.html">test_year_bool</a></span>(<span class="no">publications</span>, !<span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="no">repo</span>))</pre></body></html></div>
<pre><code>## 
## Call:
## glm(formula = bool_use ~ date_published, family = "binomial", 
##     data = pubs)
## 
## Deviance Residuals: 
##     Min       1Q   Median       3Q      Max  
## -1.4078  -0.9451  -0.5064   1.1415   1.7615  
## 
## Coefficients:
##                  Estimate Std. Error z value Pr(&gt;|z|)  
## (Intercept)    -2.451e+01  1.325e+01   -1.85   0.0643 .
## date_published  1.368e-03  7.515e-04    1.82   0.0687 .
## ---
## Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1
## 
## (Dispersion parameter for binomial family taken to be 1)
## 
##     Null deviance: 32.671  on 24  degrees of freedom
## Residual deviance: 28.156  on 23  degrees of freedom
## AIC: 32.156
## 
## Number of Fisher Scoring iterations: 4</code></pre>
<p>Maybe a bit suggestive, but not significant.</p>
</div>
<div id="number-of-genes-and-cells" class="section level1">
<h1 class="hasAnchor">
<a href="#number-of-genes-and-cells" class="anchor"></a>Number of genes and cells</h1>
<p>The mean for each study is used since I’m not always sure what counts as a dataset. Sometimes I mean one section, while sometimes I mean multiple sections the authors treated as one dataset. Actually sometimes the authors aren’t clear how many sections are there in a “dataset”, hence this confusion.</p>
<div class="sourceCode" id="cb59"><html><body><pre class="r"><span class="no">mean_genes</span> <span class="kw">&lt;-</span> <span class="no">sheet</span> <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span>(<span class="no">date_published</span>, <span class="no">title</span>, <span class="no">method</span>) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarize</a></span>(<span class="kw">n_genes</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/mean.html">mean</a></span>(<span class="no">n_genes</span>, <span class="kw">na.rm</span> <span class="kw">=</span> <span class="fl">TRUE</span>))</pre></body></html></div>
<pre><code>## `summarise()` regrouping output by 'date_published', 'title' (override with `.groups` argument)</code></pre>
<div class="sourceCode" id="cb61"><html><body><pre class="r"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="no">mean_genes</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="no">date_published</span>, <span class="no">n_genes</span>, <span class="kw">color</span> <span class="kw">=</span> <span class="no">method</span>)) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span>() +
  <span class="fu"><a href="https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html">geom_text_repel</a></span>(<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">label</span> <span class="kw">=</span> <span class="no">method</span>), <span class="kw">segment.alpha</span> <span class="kw">=</span> <span class="fl">0.5</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span>(<span class="kw">x</span> <span class="kw">=</span> <span class="st">"Date published"</span>, <span class="kw">y</span> <span class="kw">=</span> <span class="st">"Mean number of genes per study"</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(<span class="kw">legend.position</span> <span class="kw">=</span> <span class="st">"none"</span>)</pre></body></html></div>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_point).</code></pre>
<pre><code>## Warning: Removed 1 rows containing missing values (geom_text_repel).</code></pre>
<p><img src="smfish_files/figure-html/unnamed-chunk-49-1.png" width="700"></p>
<p>While the max possible number of genes increased, I don’t think people necessarily want to go for the max.</p>
<div class="sourceCode" id="cb64"><html><body><pre class="r"><span class="no">mean_genes</span> <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="no">n_genes</span> <span class="kw">&lt;</span> <span class="fl">1000</span>) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="no">date_published</span>, <span class="no">n_genes</span>)) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span>(<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">color</span> <span class="kw">=</span> <span class="no">method</span>)) +
  <span class="fu"><a href="https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html">geom_text_repel</a></span>(<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">label</span> <span class="kw">=</span> <span class="no">method</span>, <span class="kw">color</span> <span class="kw">=</span> <span class="no">method</span>), <span class="kw">segment.alpha</span> <span class="kw">=</span> <span class="fl">0.5</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_smooth.html">geom_smooth</a></span>(<span class="kw">method</span> <span class="kw">=</span> <span class="st">"lm"</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span>(<span class="kw">x</span> <span class="kw">=</span> <span class="st">"Date published"</span>, <span class="kw">y</span> <span class="kw">=</span> <span class="st">"Mean number of genes per study"</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(<span class="kw">legend.position</span> <span class="kw">=</span> <span class="st">"none"</span>)</pre></body></html></div>
<pre><code>## `geom_smooth()` using formula 'y ~ x'</code></pre>
<p><img src="smfish_files/figure-html/unnamed-chunk-50-1.png" width="700"></p>
<p>See, without those 3 outliers, there really does not seem to be an increase (if not a decrease). I wonder why people aren’t opting for the larger number of genes possible. Is it because the probes are expensive or it’s too challenging to process the sheer number of images, which is exacerbated by poor documentation of much of the existing code written for this task and lack of a unified pipeline? Now I wonder if growth of ST is in part because of STPipeline, which makes data preprocessing easier. Now I also wonder how Cell Ranger and Seurat contributed to spread of scRNA-seq in general.</p>
<p>How about number of cells? Here the total number of cells from each study per method is plotted.</p>
<div class="sourceCode" id="cb66"><html><body><pre class="r"><span class="no">sum_cells</span> <span class="kw">&lt;-</span> <span class="no">sheet</span> <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html">group_by</a></span>(<span class="no">date_published</span>, <span class="no">title</span>, <span class="no">method</span>) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html">summarize</a></span>(<span class="kw">n_cells</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(<span class="no">n_cells</span>, <span class="kw">na.rm</span> <span class="kw">=</span> <span class="fl">TRUE</span>)) <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="no">n_cells</span> <span class="kw">&gt;</span> <span class="fl">0</span>)</pre></body></html></div>
<pre><code>## `summarise()` regrouping output by 'date_published', 'title' (override with `.groups` argument)</code></pre>
<div class="sourceCode" id="cb68"><html><body><pre class="r"><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="no">sum_cells</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="no">date_published</span>, <span class="no">n_cells</span>, <span class="kw">color</span> <span class="kw">=</span> <span class="no">method</span>)) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span>() +
  <span class="fu"><a href="https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html">geom_text_repel</a></span>(<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">label</span> <span class="kw">=</span> <span class="no">method</span>), <span class="kw">segment.alpha</span> <span class="kw">=</span> <span class="fl">0.5</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span>(<span class="kw">x</span> <span class="kw">=</span> <span class="st">"Date published"</span>, <span class="kw">y</span> <span class="kw">=</span> <span class="st">"Total number of cells per study"</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(<span class="kw">legend.position</span> <span class="kw">=</span> <span class="st">"none"</span>)</pre></body></html></div>
<p><img src="smfish_files/figure-html/unnamed-chunk-52-1.png" width="700"></p>
<p>That outlier is the crazy hypothalamus MERFISH study.</p>
<div class="sourceCode" id="cb69"><html><body><pre class="r"><span class="no">sum_cells2</span> <span class="kw">&lt;-</span> <span class="no">sum_cells</span> <span class="kw">%&gt;%</span>
  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html">filter</a></span>(<span class="no">n_cells</span> <span class="kw">&lt;</span> <span class="fl">1e5</span>)
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="no">sum_cells2</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="no">date_published</span>, <span class="no">n_cells</span>)) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html">geom_point</a></span>(<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">color</span> <span class="kw">=</span> <span class="no">method</span>)) +
  <span class="fu"><a href="https://rdrr.io/pkg/ggrepel/man/geom_text_repel.html">geom_text_repel</a></span>(<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">label</span> <span class="kw">=</span> <span class="no">method</span>, <span class="kw">color</span> <span class="kw">=</span> <span class="no">method</span>), <span class="kw">segment.alpha</span> <span class="kw">=</span> <span class="fl">0.5</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_smooth.html">geom_smooth</a></span>(<span class="kw">method</span> <span class="kw">=</span> <span class="st">"lm"</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">labs</a></span>(<span class="kw">x</span> <span class="kw">=</span> <span class="st">"Date published"</span>, <span class="kw">y</span> <span class="kw">=</span> <span class="st">"Total number of cells per study"</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(<span class="kw">legend.position</span> <span class="kw">=</span> <span class="st">"none"</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/coord_cartesian.html">coord_cartesian</a></span>(<span class="kw">ylim</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>, <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span>(<span class="no">sum_cells2</span>$<span class="no">n_cells</span>)))</pre></body></html></div>
<pre><code>## `geom_smooth()` using formula 'y ~ x'</code></pre>
<p><img src="smfish_files/figure-html/unnamed-chunk-53-1.png" width="700"></p>
<p>I don’t think it’s significant. Here I formally test if beta is not 0:</p>
<div class="sourceCode" id="cb71"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span>(<span class="fu"><a href="https://rdrr.io/r/stats/lm.html">lm</a></span>(<span class="no">n_cells</span> ~ <span class="no">date_published</span>, <span class="kw">data</span> <span class="kw">=</span> <span class="no">sum_cells2</span>))</pre></body></html></div>
<pre><code>## 
## Call:
## lm(formula = n_cells ~ date_published, data = sum_cells2)
## 
## Residuals:
##    Min     1Q Median     3Q    Max 
## -10968  -8440  -6225   5740  30879 
## 
## Coefficients:
##                  Estimate Std. Error t value Pr(&gt;|t|)
## (Intercept)    -57934.772  63350.541  -0.915    0.372
## date_published      3.869      3.609   1.072    0.297
## 
## Residual standard error: 13410 on 19 degrees of freedom
## Multiple R-squared:  0.05704,    Adjusted R-squared:  0.007408 
## F-statistic: 1.149 on 1 and 19 DF,  p-value: 0.2971</code></pre>
<p>See, it’s not significant. So what does this mean? What should our priorities be? Shall we focus on in depth data analysis techniques without worrying too much about scalability yet since people aren’t opting for larger amount of data anyway, or shall we prioritize scalability since lack of scalability of data processing and analysis tools is the reason why people aren’t opting for larger amount of data? Or is that because the existing software is not very user friendly, so people consider data analysis a really daunting task? Or is it the expensive reagents and instruments and the amount of time to collect data? Or did people make up their minds and decided that it’s not necessary to profile that many cells? I don’t know. I think we can try both and see what happens in a few years. Well, given how quickly this field is evolving, I suppose months.</p>
<div class="sourceCode" id="cb73"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()</pre></body></html></div>
<pre><code>## R version 4.0.0 (2020-04-24)
## Platform: x86_64-apple-darwin17.0 (64-bit)
## Running under: macOS Catalina 10.15.5
## 
## Matrix products: default
## BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
## LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib
## 
## locale:
## [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] ggrepel_0.8.2       sf_0.9-5            dplyr_1.0.0        
## [4] purrr_0.3.4         ggplot2_3.3.2       museumst_0.0.0.9002
## 
## loaded via a namespace (and not attached):
##  [1] rnaturalearth_0.1.0     tidyr_1.1.0             viridisLite_0.3.0      
##  [4] splines_4.0.0           assertthat_0.2.1        ggtextures_0.0.1       
##  [7] sp_1.4-2                googlesheets4_0.2.0     cellranger_1.1.0       
## [10] yaml_2.2.1              urbnmapr_0.0.0.9002     pillar_1.4.6           
## [13] backports_1.1.8         lattice_0.20-41         glue_1.4.1             
## [16] digest_0.6.25           RColorBrewer_1.1-2      gridtext_0.1.1         
## [19] colorspace_1.4-1        htmltools_0.5.0.9000    Matrix_1.2-18          
## [22] pkgconfig_2.0.3         magick_2.4.0            scales_1.1.1           
## [25] tibble_3.0.3            googledrive_1.0.1       mgcv_1.8-31            
## [28] generics_0.0.2          farver_2.0.3            ellipsis_0.3.1         
## [31] withr_2.2.0             magrittr_1.5            crayon_1.3.4           
## [34] memoise_1.1.0           ggtext_0.1.0            evaluate_0.14          
## [37] tokenizers_0.2.1        janeaustenr_0.1.5       fs_1.4.2               
## [40] nlme_3.1-148            MASS_7.3-51.6           SnowballC_0.7.0        
## [43] forcats_0.5.0           xml2_1.3.2              class_7.3-17           
## [46] tools_4.0.0             gargle_0.5.0            lifecycle_0.2.0        
## [49] stringr_1.4.0           munsell_0.5.0           compiler_4.0.0         
## [52] pkgdown_1.5.1           e1071_1.7-3             rlang_0.4.7            
## [55] classInt_0.4-3          units_0.6-7             grid_4.0.0             
## [58] rstudioapi_0.11         labeling_0.3            rmarkdown_2.3          
## [61] gtable_0.3.0            DBI_1.1.0               markdown_1.1           
## [64] R6_2.4.1                lubridate_1.7.9         knitr_1.29             
## [67] rgeos_0.5-3             rprojroot_1.3-2         tidytext_0.2.5         
## [70] KernSmooth_2.23-17      desc_1.2.0              stringi_1.4.6          
## [73] Rcpp_1.0.5              vctrs_0.3.2             png_0.1-7              
## [76] wordcloud_2.6           tidyselect_1.1.0        xfun_0.16              
## [79] rnaturalearthdata_0.1.0</code></pre>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Lambda Moses, Lior Pachter.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
